{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Text summarization with spacy.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "id": "hQq2bBNcj9F0",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "import spacy"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "zH2rYoiYPTiS",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Text Preprocessing Pkg\n",
        "from spacy.lang.en.stop_words import STOP_WORDS\n",
        "from string import punctuation"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iB5NbkPUPWNA",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Build a List of Stopwords\n",
        "stopwords = list(STOP_WORDS)\n"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "uD7Nhy3PPZdd",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "with open('biomedical.txt', 'r') as file:\n",
        "    document1 = file.read().replace('\\n', '')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "gzth9354Pe58",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "with open('biomedical1.txt', 'r') as file:\n",
        "    document2 = file.read().replace('\\n', '')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bLhMqjxAPh8b",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "nlp = spacy.load('en')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "RsVr37JGPlgr",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Build an NLP Object\n",
        "docx = nlp(document1)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jfadNd9LPpq8",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Tokenization of Text\n",
        "mytokens = [token.text for token in docx]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VloBN5KuPs7_",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Build Word Frequency\n",
        "# word.text is tokenization in spacy\n",
        "word_frequencies = {}\n",
        "for word in docx:\n",
        "    if word.text not in stopwords:\n",
        "            if word.text not in word_frequencies.keys():\n",
        "                word_frequencies[word.text] = 1\n",
        "            else:\n",
        "                word_frequencies[word.text] += 1"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "KdwWkMZuPwO0",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "d9d3a6a7-6791-43f0-800c-bf9ede5753c6"
      },
      "source": [
        "\n",
        "word_frequencies"
      ],
      "execution_count": 53,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'%': 5,\n",
              " '(': 20,\n",
              " ')': 18,\n",
              " ',': 48,\n",
              " '-': 23,\n",
              " '-3α': 1,\n",
              " '.': 35,\n",
              " '/': 2,\n",
              " '1': 2,\n",
              " '128': 1,\n",
              " '130': 2,\n",
              " '146': 1,\n",
              " '159': 1,\n",
              " '165': 1,\n",
              " '2': 1,\n",
              " '21': 1,\n",
              " '22': 1,\n",
              " '32': 1,\n",
              " '39': 1,\n",
              " '43': 1,\n",
              " '5': 1,\n",
              " '60': 1,\n",
              " '61': 1,\n",
              " '95': 1,\n",
              " ';': 1,\n",
              " '<': 1,\n",
              " 'ARNT': 1,\n",
              " 'All': 2,\n",
              " 'Alterations': 1,\n",
              " 'At': 1,\n",
              " 'Besides': 1,\n",
              " 'C': 1,\n",
              " 'CBP': 2,\n",
              " 'CREB': 1,\n",
              " 'E3': 1,\n",
              " 'FIH': 4,\n",
              " 'Fig': 1,\n",
              " 'HIF': 12,\n",
              " 'HIF)-': 1,\n",
              " 'HIF-1α': 4,\n",
              " 'HIF-1β': 2,\n",
              " 'HIF-2α': 2,\n",
              " 'HIF-3α': 2,\n",
              " 'HIFs': 1,\n",
              " 'HRE': 1,\n",
              " 'Hippel': 1,\n",
              " 'Hypoxia': 3,\n",
              " 'IKK': 3,\n",
              " 'In': 5,\n",
              " 'IκB': 1,\n",
              " 'Lack': 1,\n",
              " 'Lindau': 1,\n",
              " 'MAPKs': 1,\n",
              " 'Molecular': 1,\n",
              " 'Mostly': 1,\n",
              " 'NF': 4,\n",
              " 'Nonhydroxylated': 1,\n",
              " 'Oxygen': 1,\n",
              " 'PHDs': 4,\n",
              " 'Suppression': 1,\n",
              " 'Taking': 1,\n",
              " 'The': 4,\n",
              " 'There': 1,\n",
              " 'Therefore': 2,\n",
              " 'These': 1,\n",
              " 'Under': 2,\n",
              " 'Understanding': 1,\n",
              " 'VHL': 2,\n",
              " '[': 9,\n",
              " ']': 9,\n",
              " 'accelerated': 1,\n",
              " 'activated': 2,\n",
              " 'activity': 4,\n",
              " 'acute': 3,\n",
              " 'adaptation': 2,\n",
              " 'addition': 2,\n",
              " 'alterations': 1,\n",
              " 'alveolar': 1,\n",
              " 'alveoli': 1,\n",
              " 'ambient': 1,\n",
              " 'anaerobic': 1,\n",
              " 'appears': 1,\n",
              " 'aryl': 1,\n",
              " 'asparagine': 1,\n",
              " 'associate': 1,\n",
              " 'atrium': 1,\n",
              " 'availability': 1,\n",
              " 'basis': 1,\n",
              " 'better': 1,\n",
              " 'bind': 1,\n",
              " 'binding': 1,\n",
              " 'binds': 2,\n",
              " 'blood': 2,\n",
              " 'body': 1,\n",
              " 'cancer': 2,\n",
              " 'capillary': 2,\n",
              " 'cardiac': 1,\n",
              " 'cardiovascular': 1,\n",
              " 'catalyze': 1,\n",
              " 'causing': 1,\n",
              " 'cell': 3,\n",
              " 'cells': 2,\n",
              " 'cellular': 3,\n",
              " 'center': 1,\n",
              " 'central': 1,\n",
              " 'changing': 1,\n",
              " 'chemoreceptors': 1,\n",
              " 'chronic': 4,\n",
              " 'coactivator': 1,\n",
              " 'compared': 1,\n",
              " 'complex': 2,\n",
              " 'concentration': 2,\n",
              " 'concentrations': 1,\n",
              " 'condition': 1,\n",
              " 'consideration': 1,\n",
              " 'considered': 2,\n",
              " 'constitutively': 1,\n",
              " 'consumption': 1,\n",
              " 'content': 1,\n",
              " 'continuously': 1,\n",
              " 'days': 1,\n",
              " 'death': 1,\n",
              " 'defined': 1,\n",
              " 'degradation': 4,\n",
              " 'delivery': 1,\n",
              " 'dependent': 3,\n",
              " 'deprivation': 1,\n",
              " 'designing': 1,\n",
              " 'developing': 2,\n",
              " 'diffuses': 1,\n",
              " 'dimerize': 1,\n",
              " 'direct': 1,\n",
              " 'discuss': 1,\n",
              " 'disease': 2,\n",
              " 'diseases': 4,\n",
              " 'disorders': 1,\n",
              " 'dissecting': 1,\n",
              " 'distinct': 1,\n",
              " 'distinctive': 1,\n",
              " 'domain': 1,\n",
              " 'e.g.': 2,\n",
              " 'effects': 2,\n",
              " 'electron': 1,\n",
              " 'element': 1,\n",
              " 'endometriosis': 1,\n",
              " 'endothelium': 1,\n",
              " 'energy': 1,\n",
              " 'enhance': 1,\n",
              " 'enzymes': 1,\n",
              " 'equal': 1,\n",
              " 'essentially': 1,\n",
              " 'eukaryotic': 1,\n",
              " 'example': 1,\n",
              " 'excessive': 1,\n",
              " 'exchange': 2,\n",
              " 'expressed': 3,\n",
              " 'expression': 4,\n",
              " 'expressionOxygen': 1,\n",
              " 'factor': 3,\n",
              " 'far': 1,\n",
              " 'feedback': 1,\n",
              " 'fetal': 1,\n",
              " 'focus': 1,\n",
              " 'form': 1,\n",
              " 'forms': 1,\n",
              " 'function': 3,\n",
              " 'functional': 1,\n",
              " 'functions': 2,\n",
              " 'gene': 2,\n",
              " 'general': 1,\n",
              " 'generally': 1,\n",
              " 'generated': 1,\n",
              " 'genes': 4,\n",
              " 'given': 1,\n",
              " 'glucose': 1,\n",
              " 'glycolysis': 1,\n",
              " 'harmful': 1,\n",
              " 'heart': 1,\n",
              " 'help': 1,\n",
              " 'hemoglobin': 1,\n",
              " 'heterodimeric': 1,\n",
              " 'heteroduplex': 1,\n",
              " 'homeostasis': 2,\n",
              " 'homolog': 1,\n",
              " 'hours': 1,\n",
              " 'human': 4,\n",
              " 'hydrocarbon': 1,\n",
              " 'hydrogen': 1,\n",
              " 'hydroxylases': 1,\n",
              " 'hydroxylated': 1,\n",
              " 'hydroxylates': 1,\n",
              " 'hydroxylation': 3,\n",
              " 'hypoxia': 15,\n",
              " 'hypoxic': 1,\n",
              " 'identified': 1,\n",
              " 'impacts': 1,\n",
              " 'inactivate': 1,\n",
              " 'including': 1,\n",
              " 'increase': 1,\n",
              " 'increases': 2,\n",
              " 'increasing': 1,\n",
              " 'inducible': 1,\n",
              " 'inequivalent': 1,\n",
              " 'inhibiting': 1,\n",
              " 'inhibitor': 2,\n",
              " 'inhibits': 2,\n",
              " 'initiated': 1,\n",
              " 'insufficient': 2,\n",
              " 'involves': 2,\n",
              " 'ions': 1,\n",
              " 'ischemia': 1,\n",
              " 'key': 1,\n",
              " 'kidney': 2,\n",
              " 'kinase': 1,\n",
              " 'kinases': 1,\n",
              " 'known': 1,\n",
              " 'lack': 2,\n",
              " 'lasting': 1,\n",
              " 'left': 2,\n",
              " 'level': 1,\n",
              " 'levels': 2,\n",
              " 'ligase': 1,\n",
              " 'limited': 1,\n",
              " 'long': 1,\n",
              " 'longer': 1,\n",
              " 'low': 2,\n",
              " 'lung': 2,\n",
              " 'mainly': 1,\n",
              " 'maintain': 2,\n",
              " 'maintaining': 2,\n",
              " 'manners': 1,\n",
              " 'mechanisms': 3,\n",
              " 'medulla': 1,\n",
              " 'metabolic': 1,\n",
              " 'metabolism': 1,\n",
              " 'metazoan': 1,\n",
              " 'minireview': 1,\n",
              " 'minutes': 2,\n",
              " 'mitochondrial': 1,\n",
              " 'mitogen': 1,\n",
              " 'molecular': 1,\n",
              " 'molecules': 1,\n",
              " 'months': 1,\n",
              " 'normal': 4,\n",
              " 'normoxia': 2,\n",
              " 'nuclear': 2,\n",
              " 'nucleated': 1,\n",
              " 'nucleus': 2,\n",
              " 'occur': 2,\n",
              " 'occurs': 1,\n",
              " 'order': 1,\n",
              " 'organ': 1,\n",
              " 'organisms': 2,\n",
              " 'output': 1,\n",
              " 'oxidative': 2,\n",
              " 'oxygen': 21,\n",
              " 'oxygenated': 2,\n",
              " 'p300': 2,\n",
              " 'pathological': 3,\n",
              " 'pathophysiological': 1,\n",
              " 'pathway': 2,\n",
              " 'patterns': 1,\n",
              " 'period': 1,\n",
              " 'peripheral': 1,\n",
              " 'periphery': 1,\n",
              " 'phosphorylate': 1,\n",
              " 'phosphorylated': 1,\n",
              " 'phosphorylation': 4,\n",
              " 'physiological': 2,\n",
              " 'physiologically': 1,\n",
              " 'play': 2,\n",
              " 'pons': 1,\n",
              " 'preeclampsia': 1,\n",
              " 'preexisting': 1,\n",
              " 'prevents': 3,\n",
              " 'primarily': 1,\n",
              " 'principally': 1,\n",
              " 'process': 1,\n",
              " 'processes': 1,\n",
              " 'products': 1,\n",
              " 'proline': 1,\n",
              " 'prolyl': 1,\n",
              " 'promote': 1,\n",
              " 'promotes': 1,\n",
              " 'proper': 1,\n",
              " 'proteasome': 1,\n",
              " 'protein': 6,\n",
              " 'proteins': 2,\n",
              " 'pulmonary': 2,\n",
              " 'pumps': 1,\n",
              " 'range': 1,\n",
              " 'recruitment': 2,\n",
              " 'redox': 1,\n",
              " 'reduced': 1,\n",
              " 'reduction': 1,\n",
              " 'referred': 1,\n",
              " 'referring': 1,\n",
              " 'reflecting': 1,\n",
              " 'regimens': 1,\n",
              " 'regulate': 1,\n",
              " 'regulated': 1,\n",
              " 'regulating': 1,\n",
              " 'regulation': 1,\n",
              " 'related': 1,\n",
              " 'relatively': 1,\n",
              " 'released': 1,\n",
              " 'relevancies': 1,\n",
              " 'remove': 1,\n",
              " 'required': 1,\n",
              " 'residue': 1,\n",
              " 'respiratory': 1,\n",
              " 'respond': 1,\n",
              " 'response': 3,\n",
              " 'responsive': 2,\n",
              " 'restricted': 1,\n",
              " 'result': 1,\n",
              " 'results': 1,\n",
              " 'returns': 1,\n",
              " 'role': 1,\n",
              " 'roles': 1,\n",
              " 'scavenger': 1,\n",
              " 'seconds': 1,\n",
              " 'send': 1,\n",
              " 'sense': 2,\n",
              " 'series': 1,\n",
              " 'shall': 2,\n",
              " 'short': 1,\n",
              " 'signaling': 1,\n",
              " 'signals': 1,\n",
              " 'single': 1,\n",
              " 'specialized': 1,\n",
              " 'specific': 1,\n",
              " 'stability': 1,\n",
              " 'stages': 1,\n",
              " 'state': 2,\n",
              " 'status': 1,\n",
              " 'stress': 1,\n",
              " 'stresses': 1,\n",
              " 'subunit': 1,\n",
              " 'supply': 1,\n",
              " 'suppress': 1,\n",
              " 'survival': 1,\n",
              " 'sustain': 1,\n",
              " 'switch': 1,\n",
              " 'system': 1,\n",
              " 'target': 3,\n",
              " 'tension': 1,\n",
              " 'terminal': 1,\n",
              " 'therapeutic': 1,\n",
              " 'time': 1,\n",
              " 'tissue': 2,\n",
              " 'tissues': 3,\n",
              " 'transactivation': 1,\n",
              " 'transcription': 3,\n",
              " 'transcriptional': 2,\n",
              " 'translocate': 1,\n",
              " 'translocates': 1,\n",
              " 'translocator': 1,\n",
              " 'type': 1,\n",
              " 'ubiquitin': 2,\n",
              " 'ubiquitination': 1,\n",
              " 'unit': 1,\n",
              " 'uptake': 1,\n",
              " 'usually': 1,\n",
              " 'vascular': 1,\n",
              " 'vein': 1,\n",
              " 'ventilation': 1,\n",
              " 'ventricle': 1,\n",
              " 'vessels': 1,\n",
              " 'von': 1,\n",
              " 'ways': 1,\n",
              " 'weeks': 1,\n",
              " 'wider': 1,\n",
              " 'α': 1,\n",
              " 'αs': 10,\n",
              " 'β': 1,\n",
              " 'κB': 5,\n",
              " 'κB)-': 1,\n",
              " '\\u2009': 1,\n",
              " '–': 1,\n",
              " '−2α': 1}"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 53
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "cF3krT0DPznj",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Maximum Word Frequency\n",
        "maximum_frequency = max(word_frequencies.values())\n"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "oWx49VywP4kx",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "for word in word_frequencies.keys():  \n",
        "        word_frequencies[word] = (word_frequencies[word]/maximum_frequency)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "gzIIgeS0P7vq",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "dbea75e2-434a-4598-9200-6cfe7e0e53a4"
      },
      "source": [
        "# Frequency Table\n",
        "word_frequencies"
      ],
      "execution_count": 56,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'%': 0.10416666666666667,\n",
              " '(': 0.4166666666666667,\n",
              " ')': 0.375,\n",
              " ',': 1.0,\n",
              " '-': 0.4791666666666667,\n",
              " '-3α': 0.020833333333333332,\n",
              " '.': 0.7291666666666666,\n",
              " '/': 0.041666666666666664,\n",
              " '1': 0.041666666666666664,\n",
              " '128': 0.020833333333333332,\n",
              " '130': 0.041666666666666664,\n",
              " '146': 0.020833333333333332,\n",
              " '159': 0.020833333333333332,\n",
              " '165': 0.020833333333333332,\n",
              " '2': 0.020833333333333332,\n",
              " '21': 0.020833333333333332,\n",
              " '22': 0.020833333333333332,\n",
              " '32': 0.020833333333333332,\n",
              " '39': 0.020833333333333332,\n",
              " '43': 0.020833333333333332,\n",
              " '5': 0.020833333333333332,\n",
              " '60': 0.020833333333333332,\n",
              " '61': 0.020833333333333332,\n",
              " '95': 0.020833333333333332,\n",
              " ';': 0.020833333333333332,\n",
              " '<': 0.020833333333333332,\n",
              " 'ARNT': 0.020833333333333332,\n",
              " 'All': 0.041666666666666664,\n",
              " 'Alterations': 0.020833333333333332,\n",
              " 'At': 0.020833333333333332,\n",
              " 'Besides': 0.020833333333333332,\n",
              " 'C': 0.020833333333333332,\n",
              " 'CBP': 0.041666666666666664,\n",
              " 'CREB': 0.020833333333333332,\n",
              " 'E3': 0.020833333333333332,\n",
              " 'FIH': 0.08333333333333333,\n",
              " 'Fig': 0.020833333333333332,\n",
              " 'HIF': 0.25,\n",
              " 'HIF)-': 0.020833333333333332,\n",
              " 'HIF-1α': 0.08333333333333333,\n",
              " 'HIF-1β': 0.041666666666666664,\n",
              " 'HIF-2α': 0.041666666666666664,\n",
              " 'HIF-3α': 0.041666666666666664,\n",
              " 'HIFs': 0.020833333333333332,\n",
              " 'HRE': 0.020833333333333332,\n",
              " 'Hippel': 0.020833333333333332,\n",
              " 'Hypoxia': 0.0625,\n",
              " 'IKK': 0.0625,\n",
              " 'In': 0.10416666666666667,\n",
              " 'IκB': 0.020833333333333332,\n",
              " 'Lack': 0.020833333333333332,\n",
              " 'Lindau': 0.020833333333333332,\n",
              " 'MAPKs': 0.020833333333333332,\n",
              " 'Molecular': 0.020833333333333332,\n",
              " 'Mostly': 0.020833333333333332,\n",
              " 'NF': 0.08333333333333333,\n",
              " 'Nonhydroxylated': 0.020833333333333332,\n",
              " 'Oxygen': 0.020833333333333332,\n",
              " 'PHDs': 0.08333333333333333,\n",
              " 'Suppression': 0.020833333333333332,\n",
              " 'Taking': 0.020833333333333332,\n",
              " 'The': 0.08333333333333333,\n",
              " 'There': 0.020833333333333332,\n",
              " 'Therefore': 0.041666666666666664,\n",
              " 'These': 0.020833333333333332,\n",
              " 'Under': 0.041666666666666664,\n",
              " 'Understanding': 0.020833333333333332,\n",
              " 'VHL': 0.041666666666666664,\n",
              " '[': 0.1875,\n",
              " ']': 0.1875,\n",
              " 'accelerated': 0.020833333333333332,\n",
              " 'activated': 0.041666666666666664,\n",
              " 'activity': 0.08333333333333333,\n",
              " 'acute': 0.0625,\n",
              " 'adaptation': 0.041666666666666664,\n",
              " 'addition': 0.041666666666666664,\n",
              " 'alterations': 0.020833333333333332,\n",
              " 'alveolar': 0.020833333333333332,\n",
              " 'alveoli': 0.020833333333333332,\n",
              " 'ambient': 0.020833333333333332,\n",
              " 'anaerobic': 0.020833333333333332,\n",
              " 'appears': 0.020833333333333332,\n",
              " 'aryl': 0.020833333333333332,\n",
              " 'asparagine': 0.020833333333333332,\n",
              " 'associate': 0.020833333333333332,\n",
              " 'atrium': 0.020833333333333332,\n",
              " 'availability': 0.020833333333333332,\n",
              " 'basis': 0.020833333333333332,\n",
              " 'better': 0.020833333333333332,\n",
              " 'bind': 0.020833333333333332,\n",
              " 'binding': 0.020833333333333332,\n",
              " 'binds': 0.041666666666666664,\n",
              " 'blood': 0.041666666666666664,\n",
              " 'body': 0.020833333333333332,\n",
              " 'cancer': 0.041666666666666664,\n",
              " 'capillary': 0.041666666666666664,\n",
              " 'cardiac': 0.020833333333333332,\n",
              " 'cardiovascular': 0.020833333333333332,\n",
              " 'catalyze': 0.020833333333333332,\n",
              " 'causing': 0.020833333333333332,\n",
              " 'cell': 0.0625,\n",
              " 'cells': 0.041666666666666664,\n",
              " 'cellular': 0.0625,\n",
              " 'center': 0.020833333333333332,\n",
              " 'central': 0.020833333333333332,\n",
              " 'changing': 0.020833333333333332,\n",
              " 'chemoreceptors': 0.020833333333333332,\n",
              " 'chronic': 0.08333333333333333,\n",
              " 'coactivator': 0.020833333333333332,\n",
              " 'compared': 0.020833333333333332,\n",
              " 'complex': 0.041666666666666664,\n",
              " 'concentration': 0.041666666666666664,\n",
              " 'concentrations': 0.020833333333333332,\n",
              " 'condition': 0.020833333333333332,\n",
              " 'consideration': 0.020833333333333332,\n",
              " 'considered': 0.041666666666666664,\n",
              " 'constitutively': 0.020833333333333332,\n",
              " 'consumption': 0.020833333333333332,\n",
              " 'content': 0.020833333333333332,\n",
              " 'continuously': 0.020833333333333332,\n",
              " 'days': 0.020833333333333332,\n",
              " 'death': 0.020833333333333332,\n",
              " 'defined': 0.020833333333333332,\n",
              " 'degradation': 0.08333333333333333,\n",
              " 'delivery': 0.020833333333333332,\n",
              " 'dependent': 0.0625,\n",
              " 'deprivation': 0.020833333333333332,\n",
              " 'designing': 0.020833333333333332,\n",
              " 'developing': 0.041666666666666664,\n",
              " 'diffuses': 0.020833333333333332,\n",
              " 'dimerize': 0.020833333333333332,\n",
              " 'direct': 0.020833333333333332,\n",
              " 'discuss': 0.020833333333333332,\n",
              " 'disease': 0.041666666666666664,\n",
              " 'diseases': 0.08333333333333333,\n",
              " 'disorders': 0.020833333333333332,\n",
              " 'dissecting': 0.020833333333333332,\n",
              " 'distinct': 0.020833333333333332,\n",
              " 'distinctive': 0.020833333333333332,\n",
              " 'domain': 0.020833333333333332,\n",
              " 'e.g.': 0.041666666666666664,\n",
              " 'effects': 0.041666666666666664,\n",
              " 'electron': 0.020833333333333332,\n",
              " 'element': 0.020833333333333332,\n",
              " 'endometriosis': 0.020833333333333332,\n",
              " 'endothelium': 0.020833333333333332,\n",
              " 'energy': 0.020833333333333332,\n",
              " 'enhance': 0.020833333333333332,\n",
              " 'enzymes': 0.020833333333333332,\n",
              " 'equal': 0.020833333333333332,\n",
              " 'essentially': 0.020833333333333332,\n",
              " 'eukaryotic': 0.020833333333333332,\n",
              " 'example': 0.020833333333333332,\n",
              " 'excessive': 0.020833333333333332,\n",
              " 'exchange': 0.041666666666666664,\n",
              " 'expressed': 0.0625,\n",
              " 'expression': 0.08333333333333333,\n",
              " 'expressionOxygen': 0.020833333333333332,\n",
              " 'factor': 0.0625,\n",
              " 'far': 0.020833333333333332,\n",
              " 'feedback': 0.020833333333333332,\n",
              " 'fetal': 0.020833333333333332,\n",
              " 'focus': 0.020833333333333332,\n",
              " 'form': 0.020833333333333332,\n",
              " 'forms': 0.020833333333333332,\n",
              " 'function': 0.0625,\n",
              " 'functional': 0.020833333333333332,\n",
              " 'functions': 0.041666666666666664,\n",
              " 'gene': 0.041666666666666664,\n",
              " 'general': 0.020833333333333332,\n",
              " 'generally': 0.020833333333333332,\n",
              " 'generated': 0.020833333333333332,\n",
              " 'genes': 0.08333333333333333,\n",
              " 'given': 0.020833333333333332,\n",
              " 'glucose': 0.020833333333333332,\n",
              " 'glycolysis': 0.020833333333333332,\n",
              " 'harmful': 0.020833333333333332,\n",
              " 'heart': 0.020833333333333332,\n",
              " 'help': 0.020833333333333332,\n",
              " 'hemoglobin': 0.020833333333333332,\n",
              " 'heterodimeric': 0.020833333333333332,\n",
              " 'heteroduplex': 0.020833333333333332,\n",
              " 'homeostasis': 0.041666666666666664,\n",
              " 'homolog': 0.020833333333333332,\n",
              " 'hours': 0.020833333333333332,\n",
              " 'human': 0.08333333333333333,\n",
              " 'hydrocarbon': 0.020833333333333332,\n",
              " 'hydrogen': 0.020833333333333332,\n",
              " 'hydroxylases': 0.020833333333333332,\n",
              " 'hydroxylated': 0.020833333333333332,\n",
              " 'hydroxylates': 0.020833333333333332,\n",
              " 'hydroxylation': 0.0625,\n",
              " 'hypoxia': 0.3125,\n",
              " 'hypoxic': 0.020833333333333332,\n",
              " 'identified': 0.020833333333333332,\n",
              " 'impacts': 0.020833333333333332,\n",
              " 'inactivate': 0.020833333333333332,\n",
              " 'including': 0.020833333333333332,\n",
              " 'increase': 0.020833333333333332,\n",
              " 'increases': 0.041666666666666664,\n",
              " 'increasing': 0.020833333333333332,\n",
              " 'inducible': 0.020833333333333332,\n",
              " 'inequivalent': 0.020833333333333332,\n",
              " 'inhibiting': 0.020833333333333332,\n",
              " 'inhibitor': 0.041666666666666664,\n",
              " 'inhibits': 0.041666666666666664,\n",
              " 'initiated': 0.020833333333333332,\n",
              " 'insufficient': 0.041666666666666664,\n",
              " 'involves': 0.041666666666666664,\n",
              " 'ions': 0.020833333333333332,\n",
              " 'ischemia': 0.020833333333333332,\n",
              " 'key': 0.020833333333333332,\n",
              " 'kidney': 0.041666666666666664,\n",
              " 'kinase': 0.020833333333333332,\n",
              " 'kinases': 0.020833333333333332,\n",
              " 'known': 0.020833333333333332,\n",
              " 'lack': 0.041666666666666664,\n",
              " 'lasting': 0.020833333333333332,\n",
              " 'left': 0.041666666666666664,\n",
              " 'level': 0.020833333333333332,\n",
              " 'levels': 0.041666666666666664,\n",
              " 'ligase': 0.020833333333333332,\n",
              " 'limited': 0.020833333333333332,\n",
              " 'long': 0.020833333333333332,\n",
              " 'longer': 0.020833333333333332,\n",
              " 'low': 0.041666666666666664,\n",
              " 'lung': 0.041666666666666664,\n",
              " 'mainly': 0.020833333333333332,\n",
              " 'maintain': 0.041666666666666664,\n",
              " 'maintaining': 0.041666666666666664,\n",
              " 'manners': 0.020833333333333332,\n",
              " 'mechanisms': 0.0625,\n",
              " 'medulla': 0.020833333333333332,\n",
              " 'metabolic': 0.020833333333333332,\n",
              " 'metabolism': 0.020833333333333332,\n",
              " 'metazoan': 0.020833333333333332,\n",
              " 'minireview': 0.020833333333333332,\n",
              " 'minutes': 0.041666666666666664,\n",
              " 'mitochondrial': 0.020833333333333332,\n",
              " 'mitogen': 0.020833333333333332,\n",
              " 'molecular': 0.020833333333333332,\n",
              " 'molecules': 0.020833333333333332,\n",
              " 'months': 0.020833333333333332,\n",
              " 'normal': 0.08333333333333333,\n",
              " 'normoxia': 0.041666666666666664,\n",
              " 'nuclear': 0.041666666666666664,\n",
              " 'nucleated': 0.020833333333333332,\n",
              " 'nucleus': 0.041666666666666664,\n",
              " 'occur': 0.041666666666666664,\n",
              " 'occurs': 0.020833333333333332,\n",
              " 'order': 0.020833333333333332,\n",
              " 'organ': 0.020833333333333332,\n",
              " 'organisms': 0.041666666666666664,\n",
              " 'output': 0.020833333333333332,\n",
              " 'oxidative': 0.041666666666666664,\n",
              " 'oxygen': 0.4375,\n",
              " 'oxygenated': 0.041666666666666664,\n",
              " 'p300': 0.041666666666666664,\n",
              " 'pathological': 0.0625,\n",
              " 'pathophysiological': 0.020833333333333332,\n",
              " 'pathway': 0.041666666666666664,\n",
              " 'patterns': 0.020833333333333332,\n",
              " 'period': 0.020833333333333332,\n",
              " 'peripheral': 0.020833333333333332,\n",
              " 'periphery': 0.020833333333333332,\n",
              " 'phosphorylate': 0.020833333333333332,\n",
              " 'phosphorylated': 0.020833333333333332,\n",
              " 'phosphorylation': 0.08333333333333333,\n",
              " 'physiological': 0.041666666666666664,\n",
              " 'physiologically': 0.020833333333333332,\n",
              " 'play': 0.041666666666666664,\n",
              " 'pons': 0.020833333333333332,\n",
              " 'preeclampsia': 0.020833333333333332,\n",
              " 'preexisting': 0.020833333333333332,\n",
              " 'prevents': 0.0625,\n",
              " 'primarily': 0.020833333333333332,\n",
              " 'principally': 0.020833333333333332,\n",
              " 'process': 0.020833333333333332,\n",
              " 'processes': 0.020833333333333332,\n",
              " 'products': 0.020833333333333332,\n",
              " 'proline': 0.020833333333333332,\n",
              " 'prolyl': 0.020833333333333332,\n",
              " 'promote': 0.020833333333333332,\n",
              " 'promotes': 0.020833333333333332,\n",
              " 'proper': 0.020833333333333332,\n",
              " 'proteasome': 0.020833333333333332,\n",
              " 'protein': 0.125,\n",
              " 'proteins': 0.041666666666666664,\n",
              " 'pulmonary': 0.041666666666666664,\n",
              " 'pumps': 0.020833333333333332,\n",
              " 'range': 0.020833333333333332,\n",
              " 'recruitment': 0.041666666666666664,\n",
              " 'redox': 0.020833333333333332,\n",
              " 'reduced': 0.020833333333333332,\n",
              " 'reduction': 0.020833333333333332,\n",
              " 'referred': 0.020833333333333332,\n",
              " 'referring': 0.020833333333333332,\n",
              " 'reflecting': 0.020833333333333332,\n",
              " 'regimens': 0.020833333333333332,\n",
              " 'regulate': 0.020833333333333332,\n",
              " 'regulated': 0.020833333333333332,\n",
              " 'regulating': 0.020833333333333332,\n",
              " 'regulation': 0.020833333333333332,\n",
              " 'related': 0.020833333333333332,\n",
              " 'relatively': 0.020833333333333332,\n",
              " 'released': 0.020833333333333332,\n",
              " 'relevancies': 0.020833333333333332,\n",
              " 'remove': 0.020833333333333332,\n",
              " 'required': 0.020833333333333332,\n",
              " 'residue': 0.020833333333333332,\n",
              " 'respiratory': 0.020833333333333332,\n",
              " 'respond': 0.020833333333333332,\n",
              " 'response': 0.0625,\n",
              " 'responsive': 0.041666666666666664,\n",
              " 'restricted': 0.020833333333333332,\n",
              " 'result': 0.020833333333333332,\n",
              " 'results': 0.020833333333333332,\n",
              " 'returns': 0.020833333333333332,\n",
              " 'role': 0.020833333333333332,\n",
              " 'roles': 0.020833333333333332,\n",
              " 'scavenger': 0.020833333333333332,\n",
              " 'seconds': 0.020833333333333332,\n",
              " 'send': 0.020833333333333332,\n",
              " 'sense': 0.041666666666666664,\n",
              " 'series': 0.020833333333333332,\n",
              " 'shall': 0.041666666666666664,\n",
              " 'short': 0.020833333333333332,\n",
              " 'signaling': 0.020833333333333332,\n",
              " 'signals': 0.020833333333333332,\n",
              " 'single': 0.020833333333333332,\n",
              " 'specialized': 0.020833333333333332,\n",
              " 'specific': 0.020833333333333332,\n",
              " 'stability': 0.020833333333333332,\n",
              " 'stages': 0.020833333333333332,\n",
              " 'state': 0.041666666666666664,\n",
              " 'status': 0.020833333333333332,\n",
              " 'stress': 0.020833333333333332,\n",
              " 'stresses': 0.020833333333333332,\n",
              " 'subunit': 0.020833333333333332,\n",
              " 'supply': 0.020833333333333332,\n",
              " 'suppress': 0.020833333333333332,\n",
              " 'survival': 0.020833333333333332,\n",
              " 'sustain': 0.020833333333333332,\n",
              " 'switch': 0.020833333333333332,\n",
              " 'system': 0.020833333333333332,\n",
              " 'target': 0.0625,\n",
              " 'tension': 0.020833333333333332,\n",
              " 'terminal': 0.020833333333333332,\n",
              " 'therapeutic': 0.020833333333333332,\n",
              " 'time': 0.020833333333333332,\n",
              " 'tissue': 0.041666666666666664,\n",
              " 'tissues': 0.0625,\n",
              " 'transactivation': 0.020833333333333332,\n",
              " 'transcription': 0.0625,\n",
              " 'transcriptional': 0.041666666666666664,\n",
              " 'translocate': 0.020833333333333332,\n",
              " 'translocates': 0.020833333333333332,\n",
              " 'translocator': 0.020833333333333332,\n",
              " 'type': 0.020833333333333332,\n",
              " 'ubiquitin': 0.041666666666666664,\n",
              " 'ubiquitination': 0.020833333333333332,\n",
              " 'unit': 0.020833333333333332,\n",
              " 'uptake': 0.020833333333333332,\n",
              " 'usually': 0.020833333333333332,\n",
              " 'vascular': 0.020833333333333332,\n",
              " 'vein': 0.020833333333333332,\n",
              " 'ventilation': 0.020833333333333332,\n",
              " 'ventricle': 0.020833333333333332,\n",
              " 'vessels': 0.020833333333333332,\n",
              " 'von': 0.020833333333333332,\n",
              " 'ways': 0.020833333333333332,\n",
              " 'weeks': 0.020833333333333332,\n",
              " 'wider': 0.020833333333333332,\n",
              " 'α': 0.020833333333333332,\n",
              " 'αs': 0.20833333333333334,\n",
              " 'β': 0.020833333333333332,\n",
              " 'κB': 0.10416666666666667,\n",
              " 'κB)-': 0.020833333333333332,\n",
              " '\\u2009': 0.020833333333333332,\n",
              " '–': 0.020833333333333332,\n",
              " '−2α': 0.020833333333333332}"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 56
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aF5ZYlizQBqf",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "\n",
        "# Sentence Tokens\n",
        "sentence_list = [ sentence for sentence in docx.sents ]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "R9RZv9ovQF_1",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "2de77da3-6d3a-4234-c91a-59ac268d1e1d"
      },
      "source": [
        "# Example of Sentence Tokenization,Word Tokenization and Lowering All Text\n",
        "# for t in sentence_list:\n",
        "#     for w in t:\n",
        "#         print(w.text.lower())\n",
        "[w.text.lower() for t in sentence_list for w in t ]"
      ],
      "execution_count": 58,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['in',\n",
              " 'metazoan',\n",
              " 'organisms',\n",
              " ',',\n",
              " 'the',\n",
              " 'oxygen',\n",
              " 'delivery',\n",
              " 'and',\n",
              " 'cellular',\n",
              " 'adaptation',\n",
              " 'to',\n",
              " 'oxygen',\n",
              " 'deprivation',\n",
              " 'are',\n",
              " 'accelerated',\n",
              " 'through',\n",
              " 'the',\n",
              " 'hypoxic',\n",
              " 'signaling',\n",
              " 'pathway',\n",
              " 'in',\n",
              " 'order',\n",
              " 'to',\n",
              " 'sustain',\n",
              " 'oxygen',\n",
              " 'homeostasis',\n",
              " '[',\n",
              " '130',\n",
              " ']',\n",
              " '.',\n",
              " 'lack',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'supply',\n",
              " 'or',\n",
              " 'an',\n",
              " 'excessive',\n",
              " 'oxygen',\n",
              " 'consumption',\n",
              " 'could',\n",
              " 'result',\n",
              " 'in',\n",
              " 'insufficient',\n",
              " 'oxygen',\n",
              " 'levels',\n",
              " 'for',\n",
              " 'maintaining',\n",
              " 'normal',\n",
              " 'cellular',\n",
              " 'function',\n",
              " ',',\n",
              " 'a',\n",
              " 'condition',\n",
              " 'defined',\n",
              " 'as',\n",
              " 'hypoxia',\n",
              " '.',\n",
              " 'hypoxia',\n",
              " 'may',\n",
              " 'not',\n",
              " 'be',\n",
              " 'considered',\n",
              " 'as',\n",
              " 'the',\n",
              " 'inequivalent',\n",
              " 'to',\n",
              " 'ambient',\n",
              " 'oxygen',\n",
              " 'concentration',\n",
              " '(',\n",
              " '21',\n",
              " '%',\n",
              " 'oxygen',\n",
              " ')',\n",
              " 'as',\n",
              " 'many',\n",
              " 'tissues',\n",
              " 'physiologically',\n",
              " 'function',\n",
              " 'at',\n",
              " 'levels',\n",
              " 'equal',\n",
              " 'to',\n",
              " '5',\n",
              " '%',\n",
              " 'oxygen',\n",
              " 'or',\n",
              " 'even',\n",
              " 'as',\n",
              " 'low',\n",
              " 'as',\n",
              " '1',\n",
              " '%',\n",
              " 'oxygen',\n",
              " '[',\n",
              " '61',\n",
              " ']',\n",
              " '.',\n",
              " 'mostly',\n",
              " ',',\n",
              " 'hypoxia',\n",
              " 'is',\n",
              " 'referred',\n",
              " 'to',\n",
              " 'the',\n",
              " 'relatively',\n",
              " 'low',\n",
              " '(',\n",
              " 'generally',\n",
              " '<',\n",
              " '\\u2009',\n",
              " '2',\n",
              " '%',\n",
              " ')',\n",
              " 'oxygen',\n",
              " 'content',\n",
              " 'compared',\n",
              " 'to',\n",
              " 'normal',\n",
              " 'status',\n",
              " 'in',\n",
              " 'a',\n",
              " 'given',\n",
              " 'organ',\n",
              " ',',\n",
              " 'tissue',\n",
              " ',',\n",
              " 'or',\n",
              " 'cell',\n",
              " 'type',\n",
              " '.',\n",
              " 'hypoxia',\n",
              " 'is',\n",
              " 'a',\n",
              " 'state',\n",
              " 'of',\n",
              " 'continuously',\n",
              " 'lack',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'for',\n",
              " 'a',\n",
              " 'short',\n",
              " '(',\n",
              " 'acute',\n",
              " 'hypoxia',\n",
              " ',',\n",
              " 'e.g.',\n",
              " ',',\n",
              " 'ischemia',\n",
              " ')',\n",
              " 'or',\n",
              " 'long',\n",
              " '(',\n",
              " 'chronic',\n",
              " 'hypoxia',\n",
              " ',',\n",
              " 'e.g.',\n",
              " ',',\n",
              " 'chronic',\n",
              " 'kidney',\n",
              " 'disease',\n",
              " ',',\n",
              " 'cancer',\n",
              " ')',\n",
              " 'period',\n",
              " 'of',\n",
              " 'time',\n",
              " '.',\n",
              " 'therefore',\n",
              " ',',\n",
              " 'a',\n",
              " 'wider',\n",
              " 'range',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'concentrations',\n",
              " 'and',\n",
              " 'feedback',\n",
              " 'to',\n",
              " 'acute',\n",
              " 'stresses',\n",
              " 'from',\n",
              " 'seconds',\n",
              " 'to',\n",
              " 'days',\n",
              " ',',\n",
              " 'even',\n",
              " 'weeks',\n",
              " 'to',\n",
              " 'months',\n",
              " ',',\n",
              " 'shall',\n",
              " 'be',\n",
              " 'put',\n",
              " 'into',\n",
              " 'consideration',\n",
              " 'when',\n",
              " 'referring',\n",
              " 'the',\n",
              " 'mechanisms',\n",
              " 'of',\n",
              " 'pathophysiological',\n",
              " 'relevancies',\n",
              " '.',\n",
              " 'hypoxia',\n",
              " 'is',\n",
              " 'usually',\n",
              " 'considered',\n",
              " 'to',\n",
              " 'have',\n",
              " 'pathological',\n",
              " 'effects',\n",
              " ';',\n",
              " 'however',\n",
              " ',',\n",
              " 'it',\n",
              " 'also',\n",
              " 'involves',\n",
              " 'in',\n",
              " 'maintaining',\n",
              " 'normal',\n",
              " 'physiological',\n",
              " 'functions',\n",
              " '.',\n",
              " 'taking',\n",
              " 'human',\n",
              " 'as',\n",
              " 'an',\n",
              " 'example',\n",
              " ',',\n",
              " 'the',\n",
              " 'central',\n",
              " 'and',\n",
              " 'peripheral',\n",
              " 'chemoreceptors',\n",
              " 'sense',\n",
              " 'the',\n",
              " 'reduction',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'tension',\n",
              " 'and',\n",
              " 'send',\n",
              " 'signals',\n",
              " 'to',\n",
              " 'respiratory',\n",
              " 'center',\n",
              " 'in',\n",
              " 'medulla',\n",
              " 'and',\n",
              " 'pons',\n",
              " ',',\n",
              " 'where',\n",
              " 'a',\n",
              " 'series',\n",
              " 'of',\n",
              " 'processes',\n",
              " 'are',\n",
              " 'initiated',\n",
              " 'to',\n",
              " 'increase',\n",
              " 'pulmonary',\n",
              " 'ventilation',\n",
              " 'and',\n",
              " 'cardiac',\n",
              " 'output',\n",
              " 'to',\n",
              " 'maintain',\n",
              " 'normal',\n",
              " 'functions',\n",
              " 'of',\n",
              " 'human',\n",
              " 'body',\n",
              " '.',\n",
              " 'oxygen',\n",
              " 'exchange',\n",
              " 'occurs',\n",
              " 'in',\n",
              " 'the',\n",
              " 'alveoli',\n",
              " 'of',\n",
              " 'lung',\n",
              " 'with',\n",
              " 'more',\n",
              " 'than',\n",
              " '95',\n",
              " '%',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'diffuses',\n",
              " 'into',\n",
              " 'the',\n",
              " 'capillary',\n",
              " 'vessels',\n",
              " 'via',\n",
              " 'the',\n",
              " 'alveolar',\n",
              " '-',\n",
              " 'capillary',\n",
              " 'exchange',\n",
              " 'system',\n",
              " 'and',\n",
              " 'then',\n",
              " 'binds',\n",
              " 'to',\n",
              " 'hemoglobin',\n",
              " '.',\n",
              " 'the',\n",
              " 'oxygenated',\n",
              " 'blood',\n",
              " 'returns',\n",
              " 'to',\n",
              " 'left',\n",
              " 'atrium',\n",
              " 'through',\n",
              " 'pulmonary',\n",
              " 'vein',\n",
              " '.',\n",
              " 'the',\n",
              " 'heart',\n",
              " 'then',\n",
              " 'pumps',\n",
              " 'out',\n",
              " 'the',\n",
              " 'oxygenated',\n",
              " 'blood',\n",
              " 'from',\n",
              " 'left',\n",
              " 'ventricle',\n",
              " 'to',\n",
              " 'its',\n",
              " 'periphery',\n",
              " 'to',\n",
              " 'maintain',\n",
              " 'the',\n",
              " 'proper',\n",
              " 'function',\n",
              " 'of',\n",
              " 'every',\n",
              " 'single',\n",
              " 'cell',\n",
              " '.',\n",
              " 'in',\n",
              " 'this',\n",
              " 'minireview',\n",
              " ',',\n",
              " 'we',\n",
              " 'will',\n",
              " 'not',\n",
              " 'discuss',\n",
              " 'the',\n",
              " 'physiological',\n",
              " 'effects',\n",
              " 'of',\n",
              " 'hypoxia',\n",
              " 'but',\n",
              " 'focus',\n",
              " 'on',\n",
              " 'pathological',\n",
              " 'impacts',\n",
              " 'of',\n",
              " 'hypoxia',\n",
              " 'in',\n",
              " 'several',\n",
              " 'key',\n",
              " 'human',\n",
              " 'diseases',\n",
              " 'including',\n",
              " 'cancer',\n",
              " ',',\n",
              " 'cardiovascular',\n",
              " 'diseases',\n",
              " ',',\n",
              " 'chronic',\n",
              " 'kidney',\n",
              " 'diseases',\n",
              " ',',\n",
              " 'metabolic',\n",
              " 'diseases',\n",
              " ',',\n",
              " 'preeclampsia',\n",
              " ',',\n",
              " 'and',\n",
              " 'endometriosis',\n",
              " '.',\n",
              " 'understanding',\n",
              " 'the',\n",
              " 'disease',\n",
              " 'pathological',\n",
              " 'process',\n",
              " 'shall',\n",
              " 'help',\n",
              " 'us',\n",
              " 'dissecting',\n",
              " 'the',\n",
              " 'molecular',\n",
              " 'mechanisms',\n",
              " 'of',\n",
              " 'causing',\n",
              " 'the',\n",
              " 'disorders',\n",
              " 'and',\n",
              " 'designing',\n",
              " 'better',\n",
              " 'therapeutic',\n",
              " 'regimens',\n",
              " 'against',\n",
              " 'them',\n",
              " '.',\n",
              " 'molecular',\n",
              " 'basis',\n",
              " 'of',\n",
              " 'hypoxia',\n",
              " 'in',\n",
              " 'regulating',\n",
              " 'gene',\n",
              " 'expressionoxygen',\n",
              " 'is',\n",
              " 'essentially',\n",
              " 'required',\n",
              " 'by',\n",
              " 'most',\n",
              " 'eukaryotic',\n",
              " 'organisms',\n",
              " 'as',\n",
              " 'a',\n",
              " 'scavenger',\n",
              " 'to',\n",
              " 'remove',\n",
              " 'harmful',\n",
              " 'electron',\n",
              " 'and',\n",
              " 'hydrogen',\n",
              " 'ions',\n",
              " 'generated',\n",
              " 'as',\n",
              " 'by',\n",
              " '-',\n",
              " 'products',\n",
              " 'of',\n",
              " 'mitochondrial',\n",
              " 'oxidative',\n",
              " 'phosphorylation',\n",
              " '.',\n",
              " 'at',\n",
              " 'the',\n",
              " 'cellular',\n",
              " 'level',\n",
              " ',',\n",
              " 'adaptation',\n",
              " 'involves',\n",
              " 'a',\n",
              " 'switch',\n",
              " 'of',\n",
              " 'energy',\n",
              " 'metabolism',\n",
              " 'from',\n",
              " 'oxidative',\n",
              " 'phosphorylation',\n",
              " 'to',\n",
              " 'anaerobic',\n",
              " 'glycolysis',\n",
              " ',',\n",
              " 'which',\n",
              " 'increases',\n",
              " 'glucose',\n",
              " 'uptake',\n",
              " ',',\n",
              " 'and',\n",
              " 'expression',\n",
              " 'of',\n",
              " 'stress',\n",
              " 'proteins',\n",
              " 'related',\n",
              " 'to',\n",
              " 'cell',\n",
              " 'survival',\n",
              " 'or',\n",
              " 'death',\n",
              " '[',\n",
              " '22',\n",
              " ']',\n",
              " '.',\n",
              " 'all',\n",
              " 'nucleated',\n",
              " 'cells',\n",
              " 'can',\n",
              " 'sense',\n",
              " 'oxygen',\n",
              " 'concentration',\n",
              " 'and',\n",
              " 'respond',\n",
              " 'to',\n",
              " 'reduced',\n",
              " 'oxygen',\n",
              " 'availability',\n",
              " 'in',\n",
              " 'one',\n",
              " 'of',\n",
              " 'two',\n",
              " 'distinctive',\n",
              " 'ways',\n",
              " '.',\n",
              " 'alterations',\n",
              " 'of',\n",
              " 'preexisting',\n",
              " 'proteins',\n",
              " '(',\n",
              " 'such',\n",
              " 'as',\n",
              " 'phosphorylation',\n",
              " 'or',\n",
              " 'changing',\n",
              " 'redox',\n",
              " 'state',\n",
              " ')',\n",
              " 'primarily',\n",
              " 'occur',\n",
              " 'in',\n",
              " 'response',\n",
              " 'to',\n",
              " 'acute',\n",
              " 'hypoxia',\n",
              " '(',\n",
              " 'within',\n",
              " 'minutes',\n",
              " ')',\n",
              " 'while',\n",
              " 'alterations',\n",
              " 'in',\n",
              " 'gene',\n",
              " 'expression',\n",
              " 'principally',\n",
              " 'occur',\n",
              " 'in',\n",
              " 'response',\n",
              " 'to',\n",
              " 'chronic',\n",
              " 'hypoxia',\n",
              " '(',\n",
              " 'lasting',\n",
              " 'from',\n",
              " 'minutes',\n",
              " 'to',\n",
              " 'hours',\n",
              " 'or',\n",
              " 'longer',\n",
              " ')',\n",
              " '.',\n",
              " 'the',\n",
              " 'expression',\n",
              " 'of',\n",
              " 'hypoxia',\n",
              " 'responsive',\n",
              " 'genes',\n",
              " 'is',\n",
              " 'mainly',\n",
              " 'regulated',\n",
              " 'by',\n",
              " 'hypoxia',\n",
              " 'inducible',\n",
              " 'factor',\n",
              " '(',\n",
              " 'hif)-',\n",
              " 'or',\n",
              " 'nuclear',\n",
              " 'factor',\n",
              " '-',\n",
              " 'κb',\n",
              " '(',\n",
              " 'nf',\n",
              " '-',\n",
              " 'κb)-',\n",
              " 'dependent',\n",
              " 'manners',\n",
              " '.',\n",
              " 'there',\n",
              " 'are',\n",
              " 'three',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " '(',\n",
              " 'hif-1α',\n",
              " ',',\n",
              " '−2α',\n",
              " ',',\n",
              " 'and',\n",
              " '-3α',\n",
              " ')',\n",
              " 'identified',\n",
              " 'thus',\n",
              " 'far',\n",
              " '[',\n",
              " '128',\n",
              " ']',\n",
              " '.',\n",
              " 'all',\n",
              " 'three',\n",
              " 'hifs',\n",
              " 'dimerize',\n",
              " 'with',\n",
              " 'constitutively',\n",
              " 'expressed',\n",
              " 'hif-1β',\n",
              " '(',\n",
              " 'also',\n",
              " 'known',\n",
              " 'as',\n",
              " 'aryl',\n",
              " 'hydrocarbon',\n",
              " 'nuclear',\n",
              " 'translocator',\n",
              " ',',\n",
              " 'arnt',\n",
              " ')',\n",
              " 'to',\n",
              " 'form',\n",
              " 'a',\n",
              " 'heterodimeric',\n",
              " 'functional',\n",
              " 'unit',\n",
              " '.',\n",
              " 'hif-1α',\n",
              " 'is',\n",
              " 'expressed',\n",
              " 'in',\n",
              " 'most',\n",
              " ',',\n",
              " 'if',\n",
              " 'not',\n",
              " 'all',\n",
              " ',',\n",
              " 'human',\n",
              " 'tissues',\n",
              " '[',\n",
              " '165',\n",
              " ']',\n",
              " 'while',\n",
              " 'hif-2α',\n",
              " 'and',\n",
              " 'hif-3α',\n",
              " 'are',\n",
              " 'expressed',\n",
              " 'in',\n",
              " 'more',\n",
              " 'restricted',\n",
              " 'tissues',\n",
              " 'and',\n",
              " 'developing',\n",
              " 'stages',\n",
              " 'such',\n",
              " 'as',\n",
              " 'fetal',\n",
              " 'lung',\n",
              " 'or',\n",
              " 'developing',\n",
              " 'vascular',\n",
              " 'endothelium',\n",
              " '[',\n",
              " '39',\n",
              " ',',\n",
              " '43',\n",
              " ',',\n",
              " '146',\n",
              " ']',\n",
              " '.',\n",
              " 'in',\n",
              " 'reflecting',\n",
              " 'to',\n",
              " 'their',\n",
              " 'specific',\n",
              " 'tissue',\n",
              " 'expression',\n",
              " 'patterns',\n",
              " ',',\n",
              " 'hif-1α',\n",
              " 'appears',\n",
              " 'to',\n",
              " 'play',\n",
              " 'a',\n",
              " 'general',\n",
              " 'role',\n",
              " 'in',\n",
              " 'transcriptional',\n",
              " 'regulation',\n",
              " 'of',\n",
              " 'all',\n",
              " 'cells',\n",
              " 'in',\n",
              " 'response',\n",
              " 'to',\n",
              " 'hypoxia',\n",
              " 'whereas',\n",
              " 'hif-2α',\n",
              " 'and',\n",
              " 'hif-3α',\n",
              " 'play',\n",
              " 'more',\n",
              " 'limited',\n",
              " 'or',\n",
              " 'specialized',\n",
              " 'roles',\n",
              " 'in',\n",
              " 'oxygen',\n",
              " 'homeostasis',\n",
              " '.',\n",
              " 'under',\n",
              " 'normoxia',\n",
              " ',',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'protein',\n",
              " 'are',\n",
              " 'hydroxylated',\n",
              " 'by',\n",
              " 'prolyl',\n",
              " '-',\n",
              " 'hydroxylases',\n",
              " '(',\n",
              " 'phds',\n",
              " ')',\n",
              " 'and',\n",
              " 'factor',\n",
              " 'inhibiting',\n",
              " 'hif',\n",
              " '(',\n",
              " 'fih',\n",
              " ')',\n",
              " '.',\n",
              " 'these',\n",
              " 'two',\n",
              " 'oxygen',\n",
              " '-',\n",
              " 'dependent',\n",
              " 'enzymes',\n",
              " 'are',\n",
              " 'activated',\n",
              " 'under',\n",
              " 'normoxia',\n",
              " 'and',\n",
              " 'suppress',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'activity',\n",
              " 'via',\n",
              " 'distinct',\n",
              " 'mechanisms',\n",
              " '.',\n",
              " 'phds',\n",
              " 'catalyze',\n",
              " 'the',\n",
              " 'proline',\n",
              " 'hydroxylation',\n",
              " 'of',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'so',\n",
              " 'the',\n",
              " 'e3',\n",
              " 'ubiquitin',\n",
              " 'ligase',\n",
              " ',',\n",
              " 'von',\n",
              " 'hippel',\n",
              " '–',\n",
              " 'lindau',\n",
              " '(',\n",
              " 'vhl',\n",
              " ')',\n",
              " 'can',\n",
              " 'bind',\n",
              " 'to',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'protein',\n",
              " 'and',\n",
              " 'promote',\n",
              " 'the',\n",
              " 'degradation',\n",
              " 'via',\n",
              " 'the',\n",
              " 'ubiquitin',\n",
              " 'proteasome',\n",
              " 'degradation',\n",
              " 'pathway',\n",
              " '(',\n",
              " 'fig',\n",
              " '.',\n",
              " '1',\n",
              " ')',\n",
              " '.',\n",
              " 'fih',\n",
              " 'hydroxylates',\n",
              " 'the',\n",
              " 'asparagine',\n",
              " 'residue',\n",
              " 'in',\n",
              " 'the',\n",
              " 'c',\n",
              " '-',\n",
              " 'terminal',\n",
              " 'transactivation',\n",
              " 'domain',\n",
              " 'of',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'prevents',\n",
              " 'the',\n",
              " 'recruitment',\n",
              " 'of',\n",
              " 'transcriptional',\n",
              " 'coactivator',\n",
              " 'creb',\n",
              " '-',\n",
              " 'binding',\n",
              " 'protein',\n",
              " '(',\n",
              " 'cbp',\n",
              " ')',\n",
              " 'and',\n",
              " 'its',\n",
              " 'homolog',\n",
              " ',',\n",
              " 'p300',\n",
              " ',',\n",
              " 'and',\n",
              " 'thus',\n",
              " 'inhibits',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'transcription',\n",
              " 'activity',\n",
              " '.',\n",
              " 'in',\n",
              " 'addition',\n",
              " ',',\n",
              " 'phds',\n",
              " 'and',\n",
              " 'fih',\n",
              " 'can',\n",
              " 'inactivate',\n",
              " 'nf',\n",
              " '-',\n",
              " 'κb',\n",
              " 'by',\n",
              " 'direct',\n",
              " 'hydroxylation',\n",
              " 'of',\n",
              " 'inhibitor',\n",
              " 'of',\n",
              " 'κb',\n",
              " 'kinase',\n",
              " '(',\n",
              " 'ikk',\n",
              " ')',\n",
              " 'complex',\n",
              " '[',\n",
              " '159',\n",
              " ']',\n",
              " '.',\n",
              " 'under',\n",
              " 'hypoxia',\n",
              " ',',\n",
              " 'insufficient',\n",
              " 'oxygen',\n",
              " 'inhibits',\n",
              " 'the',\n",
              " 'activity',\n",
              " 'of',\n",
              " 'phds',\n",
              " ',',\n",
              " 'thus',\n",
              " 'prevents',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'from',\n",
              " 'vhl',\n",
              " '-',\n",
              " 'dependent',\n",
              " 'protein',\n",
              " 'degradation',\n",
              " '.',\n",
              " 'in',\n",
              " 'addition',\n",
              " ',',\n",
              " 'mitogen',\n",
              " '-',\n",
              " 'activated',\n",
              " 'protein',\n",
              " 'kinases',\n",
              " '(',\n",
              " 'mapks',\n",
              " ')',\n",
              " 'phosphorylate',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " ',',\n",
              " 'which',\n",
              " 'also',\n",
              " 'increases',\n",
              " 'the',\n",
              " 'stability',\n",
              " 'of',\n",
              " 'α',\n",
              " 'subunit',\n",
              " '.',\n",
              " 'the',\n",
              " 'phosphorylated',\n",
              " 'hif',\n",
              " '-',\n",
              " 'αs',\n",
              " 'protein',\n",
              " 'translocate',\n",
              " 'to',\n",
              " 'the',\n",
              " 'nucleus',\n",
              " 'and',\n",
              " 'associate',\n",
              " 'with',\n",
              " 'hif-1β',\n",
              " ',',\n",
              " 'which',\n",
              " 'forms',\n",
              " 'a',\n",
              " 'hif-1α',\n",
              " '/',\n",
              " 'β',\n",
              " 'heteroduplex',\n",
              " 'and',\n",
              " 'binds',\n",
              " 'to',\n",
              " 'the',\n",
              " 'hypoxia',\n",
              " 'responsive',\n",
              " 'element',\n",
              " '(',\n",
              " 'hre',\n",
              " ')',\n",
              " 'of',\n",
              " 'target',\n",
              " 'genes',\n",
              " '.',\n",
              " 'suppression',\n",
              " 'of',\n",
              " 'fih',\n",
              " 'activity',\n",
              " 'under',\n",
              " 'hypoxia',\n",
              " 'results',\n",
              " 'in',\n",
              " 'increasing',\n",
              " 'cbp',\n",
              " '/',\n",
              " 'p300',\n",
              " 'recruitment',\n",
              " 'to',\n",
              " 'enhance',\n",
              " 'the',\n",
              " 'transcription',\n",
              " 'of',\n",
              " 'hif',\n",
              " 'target',\n",
              " 'genes',\n",
              " '[',\n",
              " '60',\n",
              " ',',\n",
              " '130',\n",
              " ']',\n",
              " '.',\n",
              " 'besides',\n",
              " ',',\n",
              " 'lack',\n",
              " 'of',\n",
              " 'oxygen',\n",
              " 'molecules',\n",
              " 'prevents',\n",
              " 'hydroxylation',\n",
              " 'of',\n",
              " 'ikk',\n",
              " '.',\n",
              " 'nonhydroxylated',\n",
              " 'ikk',\n",
              " 'complex',\n",
              " 'promotes',\n",
              " 'the',\n",
              " 'phosphorylation',\n",
              " ',',\n",
              " 'ubiquitination',\n",
              " ',',\n",
              " 'and',\n",
              " ...]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 58
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "S1G4lGi5QJSe",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Sentence Score via comparrng each word with sentence\n",
        "sentence_scores = {}  \n",
        "for sent in sentence_list:  \n",
        "        for word in sent:\n",
        "            if word.text.lower() in word_frequencies.keys():\n",
        "                if len(sent.text.split(' ')) < 30:\n",
        "                    if sent not in sentence_scores.keys():\n",
        "                        sentence_scores[sent] = word_frequencies[word.text.lower()]\n",
        "                    else:\n",
        "                        sentence_scores[sent] += word_frequencies[word.text.lower()]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7jFNDu9GQOUV",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 445
        },
        "outputId": "bce24e7c-a97b-4f43-c4a9-63f8733a692f"
      },
      "source": [
        "# Sentence Score Table\n",
        "sentence_scores"
      ],
      "execution_count": 60,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{In metazoan organisms, the oxygen delivery and cellular adaptation to oxygen deprivation are accelerated through the hypoxic signaling pathway in order to sustain oxygen homeostasis [130].: 3.854166666666667,\n",
              " Lack of oxygen supply or an excessive oxygen consumption could result in insufficient oxygen levels for maintaining normal cellular function, a condition defined as hypoxia.: 3.854166666666667,\n",
              " Mostly, hypoxia is referred to the relatively low (generally < 2%) oxygen content compared to normal status in a given organ, tissue, or cell type.: 5.854166666666668,\n",
              " Hypoxia is a state of continuously lack of oxygen for a short (acute hypoxia, e.g., ischemia) or long (chronic hypoxia, e.g., chronic kidney disease, cancer) period of time.: 9.333333333333334,\n",
              " Hypoxia is usually considered to have pathological effects; however, it also involves in maintaining normal physiological functions.: 2.479166666666667,\n",
              " Oxygen exchange occurs in the alveoli of lung with more than 95% of oxygen diffuses into the capillary vessels via the alveolar-capillary exchange system and then binds to hemoglobin.: 2.6041666666666665,\n",
              " The oxygenated blood returns to left atrium through pulmonary vein.: 0.9583333333333333,\n",
              " The heart then pumps out the oxygenated blood from left ventricle to its periphery to maintain the proper function of every single cell.: 1.1458333333333333,\n",
              " Understanding the disease pathological process shall help us dissecting the molecular mechanisms of causing the disorders and designing better therapeutic regimens against them.: 1.1458333333333333,\n",
              " All nucleated cells can sense oxygen concentration and respond to reduced oxygen availability in one of two distinctive ways.: 1.8541666666666665,\n",
              " The expression of hypoxia responsive genes is mainly regulated by hypoxia inducible factor (HIF)- or nuclear factor-: 1.9583333333333337,\n",
              " κB (NF-κB)- dependent manners.: 1.7083333333333335,\n",
              " There are three HIF-αs (HIF-1α, −2α, and -3α) identified thus far [128].: 4.687500000000001,\n",
              " All three HIFs dimerize with constitutively expressed HIF-1β (also known as aryl hydrocarbon nuclear translocator, ARNT) to form a heterodimeric functional unit.: 2.833333333333334,\n",
              " Under normoxia, HIF-αs protein are hydroxylated by prolyl-hydroxylases (PHDs) and factor inhibiting HIF (FIH).: 4.791666666666667,\n",
              " These two oxygen-dependent enzymes are activated under normoxia and suppress HIF-αs activity via distinct mechanisms.: 2.6875,\n",
              " In addition, PHDs and FIH can inactivate NF-κB by direct hydroxylation of inhibitor of κB kinase (IKK) complex [159].: 3.645833333333333,\n",
              " Under hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation.: 5.145833333333334,\n",
              " In addition, mitogen-activated protein kinases (MAPKs) phosphorylate HIF-αs, which also increases the stability of α subunit.: 5.0625,\n",
              " The phosphorylated HIF-αs protein translocate to the nucleus and associate with HIF-1β, which forms a HIF-1α/β heteroduplex and binds to the hypoxia responsive element (HRE) of target genes.: 4.104166666666667,\n",
              " Suppression of FIH activity under hypoxia results in increasing CBP/p300 recruitment to enhance the transcription of HIF target genes [60, 130].: 2.958333333333333,\n",
              " Besides, lack of oxygen molecules prevents hydroxylation of IKK.: 2.3541666666666665,\n",
              " Nonhydroxylated IKK complex promotes the phosphorylation, ubiquitination, and degradation of inhibitor of NF-κB (IκB).: 4.291666666666666,\n",
              " Therefore, NF-κB is released and translocates to the nucleus to regulate the transcription of target genes [32].: 2.9166666666666665}"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 60
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "mLS3tjZ8QgRi",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Import Heapq \n",
        "from heapq import nlargest"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "2wM-ohauQreJ",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "\n",
        "summarized_sentences = nlargest(7, sentence_scores, key=sentence_scores.get)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "1H5fnFTAQulP",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 156
        },
        "outputId": "89e108e2-9a6a-4b82-e7ac-17dbe6e54f17"
      },
      "source": [
        "summarized_sentences"
      ],
      "execution_count": 63,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[Hypoxia is a state of continuously lack of oxygen for a short (acute hypoxia, e.g., ischemia) or long (chronic hypoxia, e.g., chronic kidney disease, cancer) period of time.,\n",
              " Mostly, hypoxia is referred to the relatively low (generally < 2%) oxygen content compared to normal status in a given organ, tissue, or cell type.,\n",
              " Under hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation.,\n",
              " In addition, mitogen-activated protein kinases (MAPKs) phosphorylate HIF-αs, which also increases the stability of α subunit.,\n",
              " Under normoxia, HIF-αs protein are hydroxylated by prolyl-hydroxylases (PHDs) and factor inhibiting HIF (FIH).,\n",
              " There are three HIF-αs (HIF-1α, −2α, and -3α) identified thus far [128].,\n",
              " Nonhydroxylated IKK complex promotes the phosphorylation, ubiquitination, and degradation of inhibitor of NF-κB (IκB).]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 63
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "BgSDz5gRQxqp",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 156
        },
        "outputId": "c7d5c205-1a61-4ad9-bb85-3ef9d5830a13"
      },
      "source": [
        "# Convert Sentences from Spacy Span to Strings for joining entire sentence\n",
        "for w in summarized_sentences:\n",
        "    print(w.text)"
      ],
      "execution_count": 64,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Hypoxia is a state of continuously lack of oxygen for a short (acute hypoxia, e.g., ischemia) or long (chronic hypoxia, e.g., chronic kidney disease, cancer) period of time.\n",
            "Mostly, hypoxia is referred to the relatively low (generally < 2%) oxygen content compared to normal status in a given organ, tissue, or cell type.\n",
            "Under hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation.\n",
            "In addition, mitogen-activated protein kinases (MAPKs) phosphorylate HIF-αs, which also increases the stability of α subunit.\n",
            "Under normoxia, HIF-αs protein are hydroxylated by prolyl-hydroxylases (PHDs) and factor inhibiting HIF (FIH).\n",
            "There are three HIF-αs (HIF-1α, −2α, and -3α) identified thus far [128].\n",
            "Nonhydroxylated IKK complex promotes the phosphorylation, ubiquitination, and degradation of inhibitor of NF-κB (IκB).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "z0xy6_T5Q0kk",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "\n",
        "# List Comprehension of Sentences Converted From Spacy.span to strings\n",
        "final_sentences = [ w.text for w in summarized_sentences ]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ewnAqpzuQ32T",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "summary = ' '.join(final_sentences)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aUQywAXlQ6fs",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 54
        },
        "outputId": "b171cc2b-551e-468a-f643-017b541b44a6"
      },
      "source": [
        "\n",
        "summary"
      ],
      "execution_count": 67,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Hypoxia is a state of continuously lack of oxygen for a short (acute hypoxia, e.g., ischemia) or long (chronic hypoxia, e.g., chronic kidney disease, cancer) period of time. Mostly, hypoxia is referred to the relatively low (generally <\\u20092%) oxygen content compared to normal status in a given organ, tissue, or cell type. Under hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation. In addition, mitogen-activated protein kinases (MAPKs) phosphorylate HIF-αs, which also increases the stability of α subunit. Under normoxia, HIF-αs protein are hydroxylated by prolyl-hydroxylases (PHDs) and factor inhibiting HIF (FIH). There are three HIF-αs (HIF-1α, −2α, and -3α) identified thus far [128]. Nonhydroxylated IKK complex promotes the phosphorylation, ubiquitination, and degradation of inhibitor of NF-κB (IκB).'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 67
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9kQkZAqxQ_OA",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        },
        "outputId": "3f8595d1-223b-42f3-b98d-baf08a313bcd"
      },
      "source": [
        "len(summary)"
      ],
      "execution_count": 68,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "877"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 68
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "fTB76tJMRAmJ",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        },
        "outputId": "28d5f5ca-d4aa-48be-c073-31bab929db9e"
      },
      "source": [
        "len(document1)"
      ],
      "execution_count": 69,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "5812"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 69
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_58ob6ZxRDxa",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Place All As A Function For Reuseability\n",
        "def text_summarizer(raw_docx):\n",
        "    raw_text = raw_docx\n",
        "    docx = nlp(raw_text)\n",
        "    stopwords = list(STOP_WORDS)\n",
        "    # Build Word Frequency\n",
        "# word.text is tokenization in spacy\n",
        "    word_frequencies = {}  \n",
        "    for word in docx:  \n",
        "        if word.text not in stopwords:\n",
        "            if word.text not in word_frequencies.keys():\n",
        "                word_frequencies[word.text] = 1\n",
        "            else:\n",
        "                word_frequencies[word.text] += 1\n",
        "\n",
        "\n",
        "    maximum_frequncy = max(word_frequencies.values())\n",
        "\n",
        "    for word in word_frequencies.keys():  \n",
        "        word_frequencies[word] = (word_frequencies[word]/maximum_frequncy)\n",
        "    # Sentence Tokens\n",
        "    sentence_list = [ sentence for sentence in docx.sents ]\n",
        "\n",
        "    # Calculate Sentence Score and Ranking\n",
        "    sentence_scores = {}  \n",
        "    for sent in sentence_list:  \n",
        "        for word in sent:\n",
        "            if word.text.lower() in word_frequencies.keys():\n",
        "                if len(sent.text.split(' ')) < 30:\n",
        "                    if sent not in sentence_scores.keys():\n",
        "                        sentence_scores[sent] = word_frequencies[word.text.lower()]\n",
        "                    else:\n",
        "                        sentence_scores[sent] += word_frequencies[word.text.lower()]\n",
        "\n",
        "    # Find N Largest\n",
        "    summary_sentences = nlargest(7, sentence_scores, key=sentence_scores.get)\n",
        "    final_sentences = [ w.text for w in summary_sentences ]\n",
        "    summary = ' '.join(final_sentences)\n",
        "    print(\"Original Document\\n\")\n",
        "    print(raw_docx)\n",
        "    print(\"Total Length:\",len(raw_docx))\n",
        "    print('\\n\\nSummarized Document\\n')\n",
        "    print(summary)\n",
        "    print(\"Total Length:\",len(summary))"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "QVJO-t6BRUSB",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 207
        },
        "outputId": "9bb4dbba-2551-43a9-cf28-2d432d15496a"
      },
      "source": [
        "text_summarizer(document2)"
      ],
      "execution_count": 71,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Original Document\n",
            "\n",
            "Brain-derived neurotrophic factor (BDNF) is a neurotrophin that modulates neuroplasticity in the brain, and is one of the most widely investigated molecule in psychiatric disorders. The researches of BDNF emcompassed the advance of investigative techniques of past decades. BDNF researches ranged from protein quantilization, to RNA expression measurements, to DNA sequencing, and lately but not lastly, epigenetic studies. In this review, we will briefly address findings on BDNF protein levels, mRNA expression, Val66Met polymorphism, and epigenetic modifications, in schizophrenia, major depressive disorder (MDD), and bipolar disorder.Brain-derived neurotrophic factor (BDNF) is a neurotrophin that modulates neuronal survival and neuroplasticity in the brain, with important roles in the growth, survival, differentiation and repair of neurons [1]. Human BDNF gene consists of 11 exons and 9 promoters [2]. The most widely studied single nucleotide polymorphism (SNP) of BDNF gene is Val66Met polymorphism, also known as rs6265 or G196A polymorphism [3]. The biological effects of BDNF are mediated by the transmembrane tropomyosin-related kinase B (TrkB). On one hand, binding of mature BDNF to the TrkB receptor can trigger autophosphorylation of the tyrosine residue in the receptor's intracellular domain and activate downstream signaling pathways, associated with neurogenesis and neuroplasticity [4]. On the other hand, binding of pro-BDNF with p75 neurotrophin receptor mediates processes opposite to mature-BDNF, such as apoptosis, reduction in arborization, and impairment of long term potentiation [5]. Medications are also known to affect the BDNF pathway. For example, cycloporine, an immunosuppressant associated with depression, could decrease BDNF and TrkB mRNA expressions in the hippocampus and midbrain of rats [6].Past studies showed BDNF can cross the blood–brain barrier in both directions [7]. Animal studies have demonstrated that blood BDNF levels correlated positively with cortical BDNF levels [7,8]. Those studies have important implications in clinical research. While cerebrospinal fluid (CSF) and brain tissues are frequently studied in animal models, they become difficult to acquire in clinical samples, where peripheral samples such as serum and plasma are preferred.BDNF has been associated with various neuropsychiatric disorders. In this review, we will briefly address findings on BDNF protein levels, mRNA expression, Val66Met polymorphism, and epigenetic modifications, in schizophrenia, major depressive disorder (MDD), and bipolar disorder.SchizophreniaProtein and mRNAFrom serum samples of patients with schizophrenia, lower BDNF protein levels had also been reported [9]. In one of our earlier studies, we found no significant differences in serum BDNF protein levels between 126 patients with schizophrenia and 96 healthy controls, but patients with catatonic subtype had lower BDNF protein levels than patients with paranoid or residual subtype [10]. After relief of catatonia with intramuscular injections of lorazepam, patients with catatonic schizophrenia showed decreased serum BDNF levels [11]. In 34 drug-naive patients with first episode schizophrenia, significantly lower baseline BDNF protein levels were observed, as compared to healthy controls [12]. In patients with first-episode and chronic schizophrenia, lower plasma BDNF levels were observed [13]. From the plasma samples of 589 patients of schizophrenia and 254 patients with bipolar disorder, lower BDNF levels were found compared to controls; among patients, lower BDNF levels were associated with more depressive episodes, shorter time in remission, and childhood sexual abuse [14]. Those data suggested that lower peripheral BDNF levels had been relatively consistently observed in patients with schizophrenia.Given that schizophrenia was characterized by cognitive deficits in addition to psychotic symptoms, and BDNF was involved in neuroplasticity, there were studies aiming to identify their relationships. In patients with schizophrenia, we found their serum BDNF protein levels to be negatively correlated with interpersonal communication of social function and accuracy rate of conjunction search in cognitive test [15]. In patients with first-episode and chronic schizophrenia, worse cognitive performance was noted, and BDNF correlated with several cognitive dimensions [13]. A meta-analysis also found BDNF levels to be significantly associated with several cognitive domains in patients with schizophrenia [16].One thing worth noting is that gender difference of BDNF effect had been observed in schizophrenia. No difference in serum BDNF levels between patients with schizophrenia and body mass index (BMI) matched controls was initially reported, but when gender groups were analyzed separately, BDNF levels negatively correlated with BMI gain in females with schizophrenia, but not in males [17]. We found that male patients with schizophrenia with metabolic syndrome had higher serum BDNF levels than those without [18]. Pooling our data from the past ten years, we found BDNF levels of 224 patients with schizophrenia to be significantly lower than 390 controls, but comparable with levels of patients with MDD. The receiver operating characteristic curve analysis showed that BDNF levels had a moderate accuracy of differentiating female patients with schizophrenia from controls [19]. Another study found increased plasma BDNF levels in female patients with schizophrenia, as compared to male patients as well as female controls [20]. Among 132 Chinese patients with schizophrenia, female patients had lower BDNF levels compared to males [21]. Gender difference might affect the outcome of analyses of BDNF protein levels in patients with schizophrenia.In patients with schizophrenia treated with atypical antipsychotics for four weeks, their serum BDNF protein levels showed no significant change after treatment; however, serum BDNF levels were increased in the subgroup receiving risperidone compared to that receiving clozapine, but the phenomenon was observed in males only, suggesting a gender difference in terms of drug effect [22]. After six weeks of olanzapine or lurasidone treatment, unmedicated patients with schizophrenia showed increase of serum BDNF levels, and the increase was more prominent with the olanzapine group [23]. After eight weeks of olanzapine treatment, fourteen drug-naive patients with first-episode schizophrenia showed no statistical significant increase of serum BDNF levels [24]. While most data suggested an increase of BDNF levels after medications, statistical significant increase was not always achieved. Different treatment durations, different drug mechanics, and even heterogenous sample populations could all contribute to the discrepancy.Single nucleotide polymorphism (SNP)Met/Met allele of Val66Met polymorphism was initially thought to increase the risk of schizophrenia [25]. Our investigation of Val66Met polymorphism of 132 patients with schizophrenia and 103 healthy controls found no significant difference of allele or genotype distributions between the two groups, though a significant difference was detected between patients with and without suicide history [26]. A recent updated meta-analysis of 11,480 patients with schizophrenia and 13,490 healthy controls found no association between Val66Met polymorphism with schizophrenia [27]. Yet, there could be a gender and ethnic difference in how Val66Met polymorphism affect patients with schizophrenia. Female patients with schizophrenia with Val allele of the BDNF Val66Met polymorphism scored higher in PANSS scale and lower in cognitive tests, but no performance difference was observed in male patients [28]. When analyzing Val66Met polymorphism based on different ethnic groups, a meta-analysis found that Met/Met genotype was a risk factor of schizophrenia, and increased the risk of schizophrenia up to 9% in Asian, up to 26% in European populations, and up to 9% in people with Chinese origin [29]. In terms of antipsychotic response, a recent meta-analysis found no association between Val66Met polymorphism and treatment response [30]. The role of Val66Met polymorphism in schizophrenia might not be as robust as we initially thought, but the SNP may affect a range of clinical symptoms such as age of onset, treatment response, cognition, and brain morphology [31].Epigenetic modificationsEpigenetic modifications such as DNA methylation, histone modifications, and non-coding RNAs can affect gene activity and expression, without changing the DNA sequence. In a genome-wide epigenomic analysis of the postmortem brains of patients with schizophrenia and bipolar disorder, no significant methylation level difference was found in BDNF gene from controls, but Val homozygotes of Val66Met SNP in exon IX had significantly higher DNA methylation in the four nearby CpG sites than Val/Met or Met/Met carriers [32]. In another study using postmortem brains of 15 patients with schizophrenia and 15 controls, analysis of a region surrounding BDNF promoter IV revealed that a single CpG site at −93 was hypomethylated in patients with schizophrenia [33].Using DNA from blood samples of patients with schizophrenia and healthy controls, frequency of the BDNF gene methylation was highlighted as a statistically significant relationship between patients and controls regarding decreased risk of disease in comparison to unmethylated patterns [34]. In the peripheral blood cells of the patients with schizophrenia and healthy controls, patients had a higher methylation level at BDNF promoter I compared to controls, but no significant difference was detected in methylation levels of BDNF promoter IV [35].Histone methylation is the transfer to one to three methyl groups to lysine or arginine residues of histone proteins, performed by histone methyltransferase. Mixed-lineage leukemia 1 (MLL1) is a histone 3 lysine-specific methyltransferase which could affect BDNF histone modifications. We found increased levels of blood MLL1 mRNA and BDNF exon IV promoter H3K4me3 in patients with schizophrenia as compared to healthy controls [36]. In the follow-up study evaluating clozapine treatment effect, we found higher H3K9me2 levels in patients with schizophrenia, but found no difference in MLL1 mRNA levels, H3K9me2 levels, and H3K27me3 levels between patients with clozapine treatment and non-clozapine treatment [37].The most well-studied noncoding RNAs are microRNA (miRNA), which are 21–23 nucleotides long and can inhibit the translation of messenger RNA (mRNA), regulating protein expression via post-transcriptional gene silencing. A large number of miRNAs had been investigated in schizophrenia. For example, miR-183 levels in the whole blood samples were significantly decreased in the patients with tumor alone than the patients with comorbid tumor and schizophrenia [38]. The study design and investigated targets often vary widely among epigenetic studies, thus it has not been hard to find an unified conclusion. Nevertheless, as more data accumulate, we might be able to utilize those findings better in diagnosis, treatment, and prognosis.Major depressive disorder (MDD)Protein and mRNASimilar to schizophrenia, BDNF is also one of the most well studied molecule in MDD. Lower serum BDNF protein levels had been observed in patients with MDD. In our recent study, patients with MDD had a lower serum BDNF protein and mRNA levels than the healthy controls [39]. Compared to healthy controls, serum BDNF levels were lower in patients with MDD [40]. Plasma BDNF levels were significantly lower in patients with MDD and those with attempted suicide [41]. We also noted a gender difference of BDNF effect in MDD. At baseline, only female patients with MDD had significantly lower serum BDNF protein levels than controls, but no significant difference was found between male patients and controls [42]. Drug naive patients with first episode MDD had lower BDNF protein levels than healthy controls [43]. Pooling our data from the past ten years, we found BDNF levels of 224 patients with schizophrenia to be significantly lower than 390 controls, but comparable with levels of 273 patients with MDD. The receiver operating characteristic curve analysis showed that BDNF levels had a moderate accuracy of differentiating male patients with MDD from controls [19]. Generally, lower BDNF protein and mRNA levels were observed in patients with MDD.Antidepressant treatment has also been shown to increase BDNF levels [44]. A meta-analysis also reported similar findings [45]. In our recent study, we also found that BDNF protein and mRNA levels significantly increased after four weeks of antidepressant treatment [46]. Baseline plasma BDNF to leptin ratio was lower in patients with MDD, but in antidepressant responders the ratio was higher than non-responders [47]. After twelve weeks of antidepressant treatment, plasma BDNF increased significantly [41]. There may also be a gender difference in terms of BDNF and antidepressant. After antidepressant treatment, BDNF levels increased significantly, but only female responders had significantly increased changes in serum BDNF protein levels [42]. Antidepressant treatment appears to increase BDNF levels in patients with MDD.Down regulation of the TrkB signaling pathway has been suggested as one of the underlying mechanisms of MDD [48]. Patients with MDD had significantly higher TrkB levels than healthy controls [49]. Similarly, significantly higher TrkB protein levels were found in obese patients with MDD as compared to non-obese patients with MDD [50]. The other component of BDNF pathway could be susceptible to the disease process, and a holistic evaluation of the entire pathway may be needed in the future.SNPThe relationships between BDNF Val66Met polymorphism and MDD also have been extensively studied, but the results were not entirely consistent, as a meta-analysis in 2010 found no association between Val66Met polymorphism and depression, but sex-stratified analyses revealed that Met increased risk for depression in men, but not in women, hinting at a gender difference of Val66Met effect on depression [51]. Another meta-analysis found no association between Val66Met polymorphism with MDD [52], but later studies found Val66Met polymorphism moderate life stress in depression [53,54]. The findings on Val66Met polymorphism and DNA methylation levels were limited and were mostly inconsistent. Three SNPs, including Val66Met, could modulate the association between late-life depression and BDNF exon I promoter methylation levels [55]. In a study investigating patients with MDD and bipolar disorder, no significant genotype–methylation interactions were found with Val66Met polymorphism [56]. We did not find any significant difference on Val66Met methylation levels in patients with MDD [39]. A recent study found that Val66Met polymorphism was associated with less antidepressant-induced insomnia [57]. A recent prospective study found Val66Met polymorphism to have no correlation with suicide ideation or treatment response [41]. Val66Met showed no influence on serum BDNF levels and BDNF methylation levels in patients with MDD [40]. Using machine learning, Val66Met polymorphism along with several other SNPs of various genes could predict antidepressant response [58,59]. Similar to schizophrenia, the evidences of Val66Met polymorphism having a strong association with MDD appear weak, though the particular polymorphism could modulate other associated symptoms such as life stress. Furthermore, advances in deep learning might detect previously subtle effect of SNPs.Epigenetic modificationsDNA methylation studies were the most frequently investigated epigenetic studies of BDNF gene, and BDNF exon I and IV promoters were the most studied. Among 29 differentially methylated CpG sites out of 35 studied sites in BDNF exon I between patients with MDD and healthy controls, no significant difference was detected in methylation levels in BDNF exon IV [60]. D'Addario et al. reported higher methylation of BDNF exon I promoter and lower BDNF gene expression in patients with MDD [61]. In a study of both patients with MDD and bipolar disorder, patients with MDD had higher BDNF exon I promoter methylation levels than patients with bipolar disorder and healthy controls [56]. Their follow-up study found higher BDNF exon I promoter methylation levels in both patients with MDD and patients with bipolar II disorder, but not in patients with bipolar I disorder; when analysis was performed based on the mood state instead of diagnosis, the study reported higher methylation levels among patients in depressive state, than those of manic/mixed states [62]. In geriatric patients with MDD, DNA methylation levels of both BDNF promoters I and IV were associated with depression at baseline, and chronic late-life depression [55]. Three SNPs, including Val66Met polymorphism, were found to modify the association between depression and BDNF promoter I methylation level [55]. BDNF exon IV promoter methylation levels were not associated with memory performance [63]. A recent study reported increased BDNF methylation levels were detected in patients with MDD [40]. We recently found that patients with MDD had a significantly higher methylation level at CpG site 217 and lower methylation levels at CpG site 327 and CpG site 362 in BDNF exon IX than the healthy controls, along with lower BDNF protein and mRNA levels [39].Associations between BDNF methylation levels and psychiatric medications were reported. Patients treated with both mood stabilizers and antidepressants had lower BDNF exon I promoter methylation levels than those treated with just antidepressants [61]. Lack of methylation of CpG site −87 of BDNF exon IV promoter was associated with poor treatment response in patients with MDD [64]. However, in the follow-up study, the predictor function of CpG-87 of BDNF exon IV promoter was not replicated, but in a subgroup of patients with severe depression, patients with hypermethylation at CpG-87 had significantly higher remission rates than patients without a methylation [65]. In patients after acute coronary syndrome, increased BDNF methylation was associated with the finding that escitalopram was more effective than placebo for treating depressive disorder, and this effect leads to prevent persistent depressive disorder [66]. In our study, antidepressant responders had higher methylation levels at CpG site 24 and CpG site 324 of BDNF exon IX than non-responders [39].Several miRNAs had been known to affect the expression of BDNF in depression. Depressive patients had an inverse relationship between the serum BDNF levels and the miR-132/miR-182 levels [67]. Increased miR-132 levels in patients with MDD were reported, and miR-132 levels significantly correlated with visual memory [68]. We found serum miR-30e, miR-132, miR-185, and miR-212 levels were increased in patients with MDD at baseline compared to healthy controls [46].miRNAs associated with BDNF could also affect treatment response. miR-30b, miR-132, and miR-212 levels differed significantly between paroxetine-sensitive and non-sensitive cell lines [69]. After twelve weeks of escitalopram treatment in ten patients with MDD, levels of 26 miRNAs were statistically increased [70]. We found that antidepressant treatment increased serum miR-183 and miR-212 levels in patients with MDD [71]. Data on treatment response had still been scarce, however, especially among clinical populations.As in schizophrenia, epigenetic investigations of BDNF in MDD utilize a wide variety of study methods and samples, so the current findings could not be deciphered with a simple, clear message. Further studies will be needed.Bipolar disorderProtein and mRNABipolar disorder features three distinct mood states: manic, depressive, and euthymic states. Peripheral BDNF protein levels and their relationship with various mood states of bipolar disorder had been studied in the past [45,72], but not all studies were consistent [73].A meta-analysis showed that BDNF levels were decreased in patients with either depressive or manic episodes, but no significant differences were found between euthymic patients and controls [45]. Another meta-analysis reported similar findings [72]. We initially found no significant differences in serum BDNF protein levels between patients with bipolar mania and healthy controls [73]. However, in a more recent study, we did find that patients with bipolar mania had decreased BDNF protein levels compared to healthy controls [74]. Pooling our data from the past twenty years, we compared 83 patients with bipolar disorder with 222 controls matched with age, sex, and BMI and found that patients had significantly lower BDNF protein levels. Furthermore, the receiver operating characteristic curve analysis showed that BDNF levels showed a moderate accuracy of differentiating patients with bipolar disorder from healthy controls [75]. Another recent study reported lower plasma BDNF levels were found in patients with bipolar disorder [14]. Lower serum BDNF levels were observed in bipolar patients of euthymic and manic states, and BDNF levels were associated with executive functioning and verbal memory [76].Most studies concerning BDNF mRNA and bipolar disorder utilized postmortem brains. Lower BDNF mRNA levels were found in bipolar group than control group [77,78]. The patients with bipolar disorder from aforementioned studies included patients in both manic and depressive episode. In patients with only bipolar mania, we found lower BDNF mRNA levels than did the controls [74]. In another study, we simultaneously showed that patients with bipolar mania exhibited lower BDNF protein and mRNA levels than did the healthy controls [79].A meta-analysis found an increase in BDNF levels following the treatment for acute mania [72]. After about two months of treatment, the initially decreased BDNF levels of acute manic patients showed a sharp increase after medications [80]. After eight weeks of treatment, pediatric bipolar patients had significantly higher BDNF mRNA levels [81]. In contrast, our earlier work found no significant change in serum BDNF protein levels after four weeks of treatment with mood stabilizers [73], and our follow-up study found no significant change in BDNF protein and mRNA levels after four weeks of treatment [74]. Given that antipsychotics were frequently prescribed during manic phase, antipsychotics may have different effect on BDNF protein and mRNA levels. After a sixteen week follow-up, extended-release quetiapine increases BDNF levels with time in those with a depressive episode, but decreases BDNF levels with time in those with a manic/mixed episode [82]. The frequent co-prescription of antipsychotics in addition to mood stabilizer to treat acute manic patients could affect BDNF protein and mRNA levels in unexpected directions. Those data also suggest that BDNF levels could be affected by the disease state of bipolar disorder.SNPEarly findings suggested a significant association bewteen BDNF Val66Met polymorphism and bipolar disorder [83], but recent meta-analysis indicated otherwise [84,85]. When different ethnic groups were taken into consideration, Val66Met polymorphism was significantly associated with bipolar disorder in Europeans, but not in Asians [86]. A meta-analysis investigating Val66Met polymorphism's effects on cognitive functions of patients with bipolar disorder found that adult Val carriers were associated with better executive functions, verbal learning, and verbal memory, and Met carriers might negatively module the association between childhood trauma and cognitive performances, as well as structural abnormalities in cognitive cerebral structures [87]. As previously mentioned, there were gender and ethnic differences in BDNF's effect in various psychiatric disorders. More data will be needed to clarify the role of Val66Met polymorphism in bipolar disorder.Epigenetic modificationsIn the postmortem brains of 35 patients with schizophrenia and 35 patients with bipolar disorder, Val homozygotes of Val66Met polymorphism had significantly higher DNA methylation levels across the exonic region profiled with pyrosequencing [32]. In the postmortem brains of ten patients with bipolar disorder, global DNA hypermethylation and BDNF CpG hypermethylation were found [88]. In BDNF exon I, higher degree of methylation of BDNF and lower mRNA level were found in the blood of patients with bipolar II diorder, but not of patients with bipolar I disorder [89]. That study also found that patients treated with both mood stabilizers and antidepressants had increased methylation levels than the patients treated with mood stabilizers alone, and that treatment with lithium or valproate was associated with significant decrease of methylation compared to other drugs. In BDNF promoters III and V in the leukocytes of 50 patients of bipolar I disorder and 50 age and sex matched controls, five CpG sites out of 36 CpG sites showed significantly increased methylation levels [90]. In BDNF exon I, higher methylation levels were found in the blood of patients with MDD than in either bipolar disorder or control group, and higher methylation levels were associated with antidepressant treatment [56]. In a follow-up of D'Addario et al.'s work focusing on BDNF exon I using peripheral blood samples, higher methylation was found again in patients with bipolar II diosrder as compared to patients with bipolar I disorder or major depressive disorder, and higher methylation levels was found in patients with depressed state as compared to patients with manic or mixed state [62]. That study continued to find that treatment with lithium or valproate was associated with lower methylation levels in BDNF exon I. Overall, the investigated regions of BDNF methylation could vary widely between the studies, making direct comparisons challenging, not to mention the methodological and analytical differences [91]. We found that patients with bipolar mania had a higher degree of methylation at CpG site 217 and lower degree of methylation at CpG site 391 than did the healthy controls in BDNF exon IX, but no difference was detected at Val66Met polymorphisms [79]. We also found that serum BDNF protein level correlated significantly with the degree of methylation at CpG site 348, and that BDNF mRNA level correlated significantly with degree of methylation at CpG sites 134, 177, and 217.ConclusionThe researches of BDNF summarized the advance of investigative techniques of past decades. We moved from protein quantilization, to RNA expression measurements, to DNA sequencing, and lately, epigenetic studies. While BDNF had been demonstrated to be differentially expressed in many psychiatric disorders, BDNF alone could not explain everything. That leads to another line of study techniques that look for differentially expressed molecules between diseased populations and healthy controls, such as proteomics and metabolomics.Utilizing metaolomics and proteomics of medicinal plants used in sleep disorders, BDNF activity positively correlated with the isolated proteins and metabolites [92]. Proteomic approach was also utilized to identified BDNF regulated proteins in patients undergoing cochlear implantation [93]. We had also utilized proteomic investigations on psychiatric disorders, such as schizophrenia [94], MDD [[95], [96], [97]], catatonia [98], and medication effects, such as risperidone [99], antidepressants [100]. Those investigations provide a different approach which would complement our understanding of the BDNF pathway.BDNF is relatively well studied in the major psychiatric disorders, but its protein level alone could not explain everything. Advancing investigative techniques give us new insights into how BDNF behave in disease presentations as well as treatment responses. While the heterogenous nature of epigenetic studies make drawing a simple conclusion challenging, they nevertheless pave the way to a better understanding of this particular molecule, and hopefully, human brains.Conflicts of InterestThe authors declare no conflict of interest.\n",
            "Total Length: 28315\n",
            "\n",
            "\n",
            "Summarized Document\n",
            "\n",
            "We had also utilized proteomic investigations on psychiatric disorders, such as schizophrenia [94], MDD [[95], [96], [97]], catatonia [98], and medication effects, such as risperidone [99], antidepressants [100]. In this review, we will briefly address findings on BDNF protein levels, mRNA expression, Val66Met polymorphism, and epigenetic modifications, in schizophrenia, major depressive disorder (MDD), and bipolar disorder. In this review, we will briefly address findings on BDNF protein levels, mRNA expression, Val66Met polymorphism, and epigenetic modifications, in schizophrenia, major depressive disorder (MDD), and bipolar disorder. At baseline, only female patients with MDD had significantly lower serum BDNF protein levels than controls, but no significant difference was found between male patients and controls [42]. On the other hand, binding of pro-BDNF with p75 neurotrophin receptor mediates processes opposite to mature-BDNF, such as apoptosis, reduction in arborization, and impairment of long term potentiation [5]. In 34 drug-naive patients with first episode schizophrenia, significantly lower baseline BDNF protein levels were observed, as compared to healthy controls [12]. BMI) matched controls was initially reported, but when gender groups were analyzed separately, BDNF levels negatively correlated with BMI gain in females with schizophrenia, but not in males [17].\n",
            "Total Length: 1398\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "cMBQc4HdRZu_",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 54
        },
        "outputId": "dcd5d465-457e-407a-f1c9-87a1bd2030e0"
      },
      "source": [
        "document1"
      ],
      "execution_count": 72,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'In metazoan organisms, the oxygen delivery and cellular adaptation to oxygen deprivation are accelerated through the hypoxic signaling pathway in order to sustain oxygen homeostasis [130]. Lack of oxygen supply or an excessive oxygen consumption could result in insufficient oxygen levels for maintaining normal cellular function, a condition defined as hypoxia. Hypoxia may not be considered as the inequivalent to ambient oxygen concentration (21% oxygen) as many tissues physiologically function at levels equal to 5% oxygen or even as low as 1% oxygen [61]. Mostly, hypoxia is referred to the relatively low (generally <\\u20092%) oxygen content compared to normal status in a given organ, tissue, or cell type. Hypoxia is a state of continuously lack of oxygen for a short (acute hypoxia, e.g., ischemia) or long (chronic hypoxia, e.g., chronic kidney disease, cancer) period of time. Therefore, a wider range of oxygen concentrations and feedback to acute stresses from seconds to days, even weeks to months, shall be put into consideration when referring the mechanisms of pathophysiological relevancies.Hypoxia is usually considered to have pathological effects; however, it also involves in maintaining normal physiological functions. Taking human as an example, the central and peripheral chemoreceptors sense the reduction of oxygen tension and send signals to respiratory center in medulla and pons, where a series of processes are initiated to increase pulmonary ventilation and cardiac output to maintain normal functions of human body. Oxygen exchange occurs in the alveoli of lung with more than 95% of oxygen diffuses into the capillary vessels via the alveolar-capillary exchange system and then binds to hemoglobin. The oxygenated blood returns to left atrium through pulmonary vein. The heart then pumps out the oxygenated blood from left ventricle to its periphery to maintain the proper function of every single cell. In this minireview, we will not discuss the physiological effects of hypoxia but focus on pathological impacts of hypoxia in several key human diseases including cancer, cardiovascular diseases, chronic kidney diseases, metabolic diseases, preeclampsia, and endometriosis. Understanding the disease pathological process shall help us dissecting the molecular mechanisms of causing the disorders and designing better therapeutic regimens against them.Molecular basis of hypoxia in regulating gene expressionOxygen is essentially required by most eukaryotic organisms as a scavenger to remove harmful electron and hydrogen ions generated as by-products of mitochondrial oxidative phosphorylation. At the cellular level, adaptation involves a switch of energy metabolism from oxidative phosphorylation to anaerobic glycolysis, which increases glucose uptake, and expression of stress proteins related to cell survival or death [22]. All nucleated cells can sense oxygen concentration and respond to reduced oxygen availability in one of two distinctive ways. Alterations of preexisting proteins (such as phosphorylation or changing redox state) primarily occur in response to acute hypoxia (within minutes) while alterations in gene expression principally occur in response to chronic hypoxia (lasting from minutes to hours or longer). The expression of hypoxia responsive genes is mainly regulated by hypoxia inducible factor (HIF)- or nuclear factor-κB (NF-κB)- dependent manners. There are three HIF-αs (HIF-1α, −2α, and -3α) identified thus far [128]. All three HIFs dimerize with constitutively expressed HIF-1β (also known as aryl hydrocarbon nuclear translocator, ARNT) to form a heterodimeric functional unit. HIF-1α is expressed in most, if not all, human tissues [165] while HIF-2α and HIF-3α are expressed in more restricted tissues and developing stages such as fetal lung or developing vascular endothelium [39, 43, 146]. In reflecting to their specific tissue expression patterns, HIF-1α appears to play a general role in transcriptional regulation of all cells in response to hypoxia whereas HIF-2α and HIF-3α play more limited or specialized roles in oxygen homeostasis. Under normoxia, HIF-αs protein are hydroxylated by prolyl-hydroxylases (PHDs) and factor inhibiting HIF (FIH). These two oxygen-dependent enzymes are activated under normoxia and suppress HIF-αs activity via distinct mechanisms. PHDs catalyze the proline hydroxylation of HIF-αs so the E3 ubiquitin ligase, von Hippel–Lindau (VHL) can bind to HIF-αs protein and promote the degradation via the ubiquitin proteasome degradation pathway (Fig. 1). FIH hydroxylates the asparagine residue in the C-terminal transactivation domain of HIF-αs prevents the recruitment of transcriptional coactivator CREB-binding protein (CBP) and its homolog, p300, and thus inhibits HIF-αs transcription activity. In addition, PHDs and FIH can inactivate NF-κB by direct hydroxylation of inhibitor of κB kinase (IKK) complex [159]. Under hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation. In addition, mitogen-activated protein kinases (MAPKs) phosphorylate HIF-αs, which also increases the stability of α subunit. The phosphorylated HIF-αs protein translocate to the nucleus and associate with HIF-1β, which forms a HIF-1α/β heteroduplex and binds to the hypoxia responsive element (HRE) of target genes. Suppression of FIH activity under hypoxia results in increasing CBP/p300 recruitment to enhance the transcription of HIF target genes [60, 130]. Besides, lack of oxygen molecules prevents hydroxylation of IKK. Nonhydroxylated IKK complex promotes the phosphorylation, ubiquitination, and degradation of inhibitor of NF-κB (IκB). Therefore, NF-κB is released and translocates to the nucleus to regulate the transcription of target genes [32].'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 72
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Lhk1nV01RpgO",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Get Total Word Counts with Tokenization\n",
        "docx1 = nlp(document1)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "OL0ElgUlRtUn",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Tokens\n",
        "mytokens = [ token.text for token in docx1 ]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "oqaSjWqbRxAn",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        },
        "outputId": "887fe0ef-c366-4c4f-8d98-8eb418b01ce5"
      },
      "source": [
        "# Total Number or Length of Words\n",
        "len(mytokens)"
      ],
      "execution_count": 75,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1034"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 75
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Ma6wfCZyRzqr",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# Reading Time\n",
        "def readingTime(docs):\n",
        "    total_words_tokens =  [ token.text for token in nlp(docs)]\n",
        "    estimatedtime  = len(total_words_tokens)/200\n",
        "    return '{} mins'.format(round(estimatedtime))"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-Sn12nOYR2Om",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        },
        "outputId": "93a60101-e0ef-4cf0-cf27-2273896ce05b"
      },
      "source": [
        "readingTime(document1)"
      ],
      "execution_count": 77,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'5 mins'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 77
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9bJphinyR8UW",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "from gensim.summarization import summarize"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "CgxjdUbPSHHs",
        "colab_type": "code",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 54
        },
        "outputId": "21f40e18-4cac-44d6-e46a-e55f70259638"
      },
      "source": [
        "summarize(document1)"
      ],
      "execution_count": 79,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Therefore, a wider range of oxygen concentrations and feedback to acute stresses from seconds to days, even weeks to months, shall be put into consideration when referring the mechanisms of pathophysiological relevancies.Hypoxia is usually considered to have pathological effects; however, it also involves in maintaining normal physiological functions.\\nIn reflecting to their specific tissue expression patterns, HIF-1α appears to play a general role in transcriptional regulation of all cells in response to hypoxia whereas HIF-2α and HIF-3α play more limited or specialized roles in oxygen homeostasis.\\nFIH hydroxylates the asparagine residue in the C-terminal transactivation domain of HIF-αs prevents the recruitment of transcriptional coactivator CREB-binding protein (CBP) and its homolog, p300, and thus inhibits HIF-αs transcription activity.\\nUnder hypoxia, insufficient oxygen inhibits the activity of PHDs, thus prevents HIF-αs from VHL-dependent protein degradation.\\nThe phosphorylated HIF-αs protein translocate to the nucleus and associate with HIF-1β, which forms a HIF-1α/β heteroduplex and binds to the hypoxia responsive element (HRE) of target genes.\\nSuppression of FIH activity under hypoxia results in increasing CBP/p300 recruitment to enhance the transcription of HIF target genes [60, 130].'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 79
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "O9m4EBfqSSTL",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        ""
      ],
      "execution_count": 0,
      "outputs": []
    }
  ]
}